<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your blog or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title><center> In vivo transcriptional of injected and infected macrophages <center> – DIY.transcriptomics</title>
	<meta name="description" content="">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="<center> In vivo transcriptional of injected and infected macrophages <center> – DIY.transcriptomics">
	<meta name="twitter:description" content="">
	<meta name="twitter:image:src" content="http://localhost:4000">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="<center> In vivo transcriptional of injected and infected macrophages <center> – DIY.transcriptomics" />
	<meta property="og:description" content="" />
	<meta property="og:image" content="http://localhost:4000" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/Code/files/Rmarkdown_notebook_template.Rmd">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Syllabus</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="<center> In vivo transcriptional of injected and infected macrophages <center> – DIY.transcriptomics">

			<section class="hero hero--single">

	<div class="hero__image" style="background-image: url()">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1><center> In vivo transcriptional of injected and infected macrophages <center></h1>
		<p></p>

	</div>

</section>

<section class="single">

	<div class="wrap">

		<article class="single-post">

			This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

# Abstract
This document contains all the code used to analyze the unpublished data from the ....  

# R packages
**These are the R/bioconductor packages used for this analysis:**
```{r packages, message=FALSE, warning=FALSE, results='hide'}
library(dplyr)
library(reshape2)
library(as_tibble)
library(ggplot2)
library(ggvis)
library(GSEABase)
library(GSVA)
library(reshape2)
library(readr)
library(Biostrings)
library(tximport)
library(ensembldb)
library(EnsDb.Mmusculus.v79)
library(DT)
library(limma)
library(edgeR)
library(gplots) 
library(RColorBrewer)
library(scatterD3)
library(STRINGdb)
library(patchwork)
```

**This dynamic html summary report was compiled in Rmarkdown using the following packages:**
```{r packages.continued, results='hide', message=FALSE, warning=FALSE}
library(rmarkdown)
library(knitr) 
```

# Data generation
## read mapping with Kallisto
```{r read mapping, message=FALSE, warning=FALSE, echo=FALSE, eval=FALSE}
# paste you code for aligning reads here
```

## read mapping summary
```{r mapping stats, message=FALSE, warning=FALSE}
# create a graph of your read mapping stats if you want
```


## annotation with Ensembl
```{r annotation, message=FALSE, warning=FALSE, eval=FALSE}
# all the code for anootating transcripts (Biomart or Ensembl package)
```

## importing data into R
```{r data import, message=FALSE, warning=FALSE, eval=FALSE}
# importing data into R using Tximport package
```

# Preprocessing and normalization
## study design
```{r design, message=FALSE, warning=FALSE}
#read in your study design file from local directory
```

## raw data
```{r raw, message=FALSE, warning=FALSE}
#read in your Txi_gene object and graph raw data
```

## filtered data
```{r filtering, message=FALSE, warning=FALSE}
#filter out lowly expressed genes or transcripts and regraph
```

## normalized data
```{r normalize, message=FALSE, warning=FALSE}
#normalize data using 'calcNormFactors' function, then regraph
```

# Exploratory data analysis
## Principal component analysis (PCA)
```{r PCA, message=FALSE, warning=FALSE}
#carry out PCA of filtered normalized data and graph
```


# Differential gene analysis
## set up model matrix
```{r model, message=FALSE, warning=FALSE}
#set-up your experimental design using the model.matrix
```

## mean-variance and linear model
```{r linear model, message=FALSE, warning=FALSE}
#use Limma VOOM to model mean-vairance trend
```

## set up contrasts
```{r contrasts, message=FALSE, warning=FALSE}
#fit linear model to deata and set-up contrast matrix with pairwise comparisons of interest
```

## Bayesan stats
```{r Bayes, message=FALSE, warning=FALSE}
#extract bayesian stats for linear model fit
```

## Top 10 DEGS
```{r topTable, message=FALSE, warning=FALSE}
#display top N genes for a single pairwise comparison 
#insert additional code chuncks for additional pairwise comparisons
```

## Volcano plot - infected vs double neg
```{r volcano1, message=FALSE, warning=FALSE}
#make volcano plot of pairwise comparison of interest
#insert additional code chuncks for additional pairwise comparisons=
```

## Venn
```{r Venn, message=FALSE, warning=FALSE}
#run 'decideTests' function and plot results as Venn diagram
```

## DEGs
```{r DEGs, message=FALSE, warning=FALSE}
#extract DEGs and create data frame with group averages and logFCs.  Display as table
```

# Visualizing DEGs
## heatmap of all DEGs 
```{r heatmap, message=FALSE, warning=FALSE}
#heatmap of all DEGs
#create additional heatmaps for co-expressed modules or a priori list of genes
```

# Functional enrichment analysis
## Summary of GO enrichment analysis of DEGs
```{r GO bubble, message=FALSE, warning=FALSE}
# Read in text file with results of GO enrichment carried out on DAVID or other websites
```

## GSEA with CAMERA
```{r GSEA, message=FALSE, warning=FALSE, eval= FALSE, echo=FALSE}
# carry out GSEA
```

# Session Info
**Session Info:**
```{r sessionInfo, results='asis', message=FALSE, warning=FALSE, echo=FALSE}
sessionInfo()
```



		</article>

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">A course on using open-source tools to analyze RNAseq data.</p>

		<div class="footer__copyright">
			<span>© 2019 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	


</body>

</html>