---
title: "<center> Supplementary Code for _the best class ever!_ <center>"
author: "<center> Daniel Beiting and Corbett Berry <center><br>"
date: "<center> _April, 2017_ <center>"
output:
  html_document:
    toc: yes
  pdf_document:
    fig_caption: yes
    toc: yes
header-includes:
- \usepackage{fancyhdr}
- \pagestyle{fancy}
- \lhead{Supplementary code - my awesome class}
- \chead{}
- \rhead{}
- \lfoot{Beiting et al.}
- \cfoot{}
- \rfoot{\thepage}
- \renewcommand{\headrulewidth}{0.4pt}
- \renewcommand{\footrulewidth}{0.4pt}
---

# Abstract
This document contains all the code used to analyze the unpublished data from the infection of bulk cultures of HCT-8 cells with Cryptosporidium.  Raw data is available on the Gene Expression Omnibus (GEO) under accession [GSE67136](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67136). 

# R packages
**These are the R/bioconductor packages used for this analysis:**
```{r packages, message=FALSE, warning=FALSE, results='hide'}
library(dplyr)
library(ggplot2)
library(reshape2)
library(readr)
library(Biostrings)
library(tximport)
library(ensembldb)
library(EnsDb.Hsapiens.v79)
library(ggvis)
library(DT)
library(limma)
library(edgeR)
library(gplots) 
library(RColorBrewer)
library(scatterD3)
library(rgl)
library(pca3d)
```

**This dynamic html summary report was compiled in Rmarkdown using the following packages:**
```{r packages.continued, results='hide', message=FALSE, warning=FALSE}
library(rmarkdown)
library(knitr) 
```

# Exploratory data analysis
## hierarcical clustering
```{r hierarchical clustering, message=FALSE, warning=FALSE}
load("Txi_gene")
myTPM <- Txi_gene$abundance
targets <- read_tsv("Crypto_studyDesign.txt", row.names=NULL, header = T, as.is = T)
groups <- targets$treatment
groups <- factor(groups)
rep <- factor(targets$rep)
labels <- paste(targets$treatment, targets$rep, sep = ".")

distance <- dist(t(myTPM), method="maximum") #other dist methods are "maximum", "manhattan", "canberra", "binary" or "minkowski"
clusters <- hclust(distance, method = "complete") #other methods are ward.D, ward.D2, single, complete, average
plot(clusters, labels=labels)
```

## Principal component analysis (PCA)
```{r PCA, message=FALSE, warning=FALSE}
pca.res <- prcomp(t(myTPM), scale.=F, retx=T)
pc.var<-pca.res$sdev^2 #sdev^2 gives you the eigenvalues
pc.per<-round(pc.var/sum(pc.var)*100, 1)
data.frame <- as.data.frame(pca.res$x)
ggplot(data.frame, aes(x=PC1, y=PC2, color=groups)) +
  geom_point(size=5) +
  theme(legend.position="right")
```

## Small multiples
```{r PCA-small multiples, message=FALSE, warning=FALSE}
melted <- cbind(rep, melt(pca.res$x[,1:4]))
ggplot(melted) +
  geom_bar(aes(x=Var1, y=value, fill=rep), stat="identity") +
  facet_wrap(~Var2)
```

# Differential expression analysis
## normalization
```{r finding DEGs, message=FALSE, warning=FALSE}
myTPM_raw <- Txi_gene$abundance
myTPM_filt <- myTPM_raw[rowSums(Txi_gene$counts)>=10,]
counts_filt <- Txi_gene$counts[rowSums(Txi_gene$counts)>=10,]
myDGEList <- DGEList(counts_filt)

#groups <- relevel(groups, "control") #may need to use 'relevel' function
design <- model.matrix(~0 + groups)
colnames(design) <- levels(groups)

#normalize your data using the mean-variance relationship using the VOOM function from Limma
normData <- voom(myDGEList, design, plot = TRUE)

```

## linear model fit and comparisons
```{r linear model, message=FALSE, warning=FALSE}
# fit a linear model to your data
fit <- lmFit(normData, design)
# set up a contrast matrix based on the pairwise comparisons of interest
contrast.matrix <- makeContrasts(WT_crypto_infection = wt_crypto - control,
                                 trans_crypto_infection = trans_crypto - control,
                                 levels=design)

# extract the linear model fit for the contrast matrix that you just defined above
fits <- contrasts.fit(fit, contrast.matrix)
#get bayesian stats for your linear model fit
ebFit <- eBayes(fits)
```

## volcano plot
```{r volcano plot, message=FALSE, warning=FALSE}
myTopHits <- topTable(ebFit, adjust ="BH", coef=2, number=20000, sort.by="logFC")
# Make a basic volcano plot
with(myTopHits, plot(logFC, -log10(adj.P.Val), pch=20, main="Volcano plot"))
with(subset(myTopHits, adj.P.Val<.05 & abs(logFC)>0.59), points(logFC, -log10(adj.P.Val), pch=20, col="red"))
```

## table of DEGs
```{r toptable, message=FALSE, warning=FALSE}
myTopHits <- topTable(ebFit, adjust ="BH", coef=2, number=10, sort.by="logFC")
datatable(myTopHits, 
          extensions = c('KeyTable', "FixedHeader"), 
          caption = 'Table 1: DEGs for infected (wt Crypto) vs control',
          options = list(keys = TRUE, searchHighlight = TRUE, pageLength = 10, lengthMenu = c("10", "25", "50", "100"))) %>%
  formatRound(columns=c(1:6), digits=3)
```

## Venn
```{r Venn, message=FALSE, warning=FALSE}
results <- decideTests(ebFit, method="global", adjust.method="BH", p.value=0.01, lfc=1)
vennDiagram(results, include="down")
colnames(normData$E) <- labels
diffGenes <- normData$E[results[,1] !=0 | results[,2] !=0,]
```

## heatmap
make heatmap from the `r length((diffGenes)[,1])` differentially expressed genes that were identified above. _This heatmap appears in Figure 5a of the manuscript_
```{r heatmap, message=FALSE, warning=FALSE}
#color pallette
myheatcol <- colorRampPalette(colors=c("yellow","white","blue"))(100)

#begin by clustering the genes (rows) in each set of differentially expressed genes
hr <- hclust(as.dist(1-cor(t(diffGenes), method="pearson")), method="complete") 

#now cluster your samples (columns)
#we may not acutally use this clustering result, but it's good to have just in case
hc <- hclust(as.dist(1-cor(diffGenes, method="spearman")), method="complete") 

# Cut the resulting tree and create color vector for clusters.  
mycl <- cutree(hr, k=2)

#now assign a color to each cluster (makes it easy to identify and manipulate)
mycolhc <- rainbow(length(unique(mycl)), start=0.1, end=0.9) 
mycolhc <- mycolhc[as.vector(mycl)] 

#plot the hclust results as a heatmap
# first, a heatmap of genes regulated by LPS in a WT background
heatmap.2(diffGenes, Rowv=as.dendrogram(hr), Colv=NA, 
          col=myheatcol, scale="row", labRow=NA,
          density.info="none", trace="none", RowSideColors=mycolhc, 
          cexRow=1, cexCol=1, margins=c(8,22))
```

## heatmap of genes downregulated by infection
```{r downgenes from cluster 1, fig.width=8, fig.height=6, message=FALSE, warning=FALSE}
clid <- c(1)
ysub <- diffGenes[names(mycl[mycl%in%clid]),] 
hrsub <- hclust(as.dist(1-cor(t(ysub), method="pearson")), method="complete") 
clusterIDs <- data.frame(Labels=rev(hrsub$labels[hrsub$order]))
clusterIDs <- as.vector(t(clusterIDs))

# Create heatmap for chosen sub-cluster.
heatmap.2(ysub, Rowv=as.dendrogram(hrsub), Colv=NA, col=myheatcol, scale="row", 
          density.info="none", trace="none", 
          RowSideColors=mycolhc[mycl%in%clid], margins=c(8,20)) 

```

## heatmap of genes upregulated by infection
```{r upgenes from cluster 1, fig.width=8, fig.height=6, message=FALSE, warning=FALSE}
clid <- c(2)
ysub <- diffGenes[names(mycl[mycl%in%clid]),] 
hrsub <- hclust(as.dist(1-cor(t(ysub), method="pearson")), method="complete") 
clusterIDs <- data.frame(Labels=rev(hrsub$labels[hrsub$order]))
clusterIDs <- as.vector(t(clusterIDs))

# Create heatmap for chosen sub-cluster.
heatmap.2(ysub, Rowv=as.dendrogram(hrsub), Colv=NA, col=myheatcol, scale="row", 
          density.info="none", trace="none", 
          RowSideColors=mycolhc[mycl%in%clid], margins=c(8,20)) 

```

# Session Info
**Session Info:**
```{r sessionInfo, results='asis', message=FALSE, warning=FALSE, echo=FALSE}
sessionInfo()
```
