# Introduction to this script ----
# for the purposes of this script we'll want several objects generated by the previous scripts:
#1) your normalized expression data, in the form of a data matrix, symbols as rownames
#2) your study design file
#3) your contrast matrix that lays out the pairwise comparison you're interested in
#4) Gene sets that you'd like to test for enrichment in your array data
#these gene sets can be 'custom' made based on your specific interests,
#or they can be downloaded from gene signature databases such as MSigDB or GeneSignatureDB

#generally speaking, there are two approaches for gene set testing. 
#1) Self-contained GSEA: tests whether genes in a signature are differentially expressed, in an absolute sense, without regard to any other genes in your data
#2) Competitive GSEA: tests whether genes in a signature are enriched when taking into consideration the rest of your data
#below is some code for both approaches

# Load packages ----
library(GSEABase)
library(GSVA)
library(Biobase)
library(limma)

# OPTIONAL: convert symbols to upper case 'human' ----
# make sure to have gene symbols in all caps (coerced to human)
# if working with mouse data, use the 'toupper' function to conver to uppercase (i.e. human genes)
# of course, if you're working with human RNAseq data, you won't need to do this bit
upperSymbols <- as.data.frame(toupper(rownames(normData$E))) #tolower to convert symbols to mouse
head(upperSymbols)
upperSymbols <- as.matrix(upperSymbols)
rownames(normData$E) <- upperSymbols
head(normData$E)
#sometimes there's a row with no symbol identifier...if so, go ahead an remove this
normData$E <- normData$E[-1,]

# OPTIONAL: check for and remove any duplicate rows ---- 
dups <- base::duplicated(rownames(normData$E))
unique(rownames(normData$E)[duplicated(rownames(normData$E))])
collapsed.matrix <- normData$E[!duplicated(rownames(normData$E)), ]

# load MSigDB signatures -----
#You'll need to download the .GMT files to your computer directly from from the MSigDB website (I choose the gene symbol files).  
#These can be found at:  http://www.broadinstitute.org/gsea/msigdb/collections.jsp
#place these files in a folder on your computer
#the first part of this script points to these files, so be sure to change these lines to match the directory on your computer.
# NOTE: all MSigDB files are HUMAN genes, so you may not want to query a mouse dataset against them
# if you're working with mouse array data, use MSigDB files that have been mapped to human/mouse orthologs. These can be found at:
# http://bioinf.wehi.edu.au/software/MSigDB/
# NOTE: in addition to the gene sets above, 
# you can also load custom sets (as .gmt files) downloaded from MSigDB search results (i.e. all sets with 'Treg' in the description)

#MSigDB set C2 contains pathways and chemical/genetic perturbations
broadSet.C2.ALL <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c2.all.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C2.CP <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c2.cp.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C2.CGP <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c2.cgp.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
# smaller subsets of C2
broadSet.C2.KEGG <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c2.cp.kegg.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C2.Biocarta <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c2.cp.biocarta.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C2.Reactome <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c2.cp.reactome.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
#set C3 contains lists of targets for of TFs and MIRs
broadSet.C3.ALL <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c3.all.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C3.TFT <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c3.tft.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C3.MIR <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c3.mir.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
#set C5 contains sets of genes associated with different GO terms
broadSet.C5.ALL <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c5.all.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C5.BP <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c5.bp.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
broadSet.C5.MF <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c5.mf.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())
#MsigDB set C7 contains list of genes associated with immunological signatures
broadSet.C7 <- getGmt("/Users/danielbeiting/Dropbox/MSigDB/c7.all.v5.0.symbols.gmt", geneIdType=SymbolIdentifier())

# self-contained GSEA using ROAST----
# first let's creat a few signatures to test in our enrichment analysis
mySig <- toupper(rownames(myTopHits))
mySig2 <- c("AAK1", "AADAT", "AAGAB", "AADACL4", "AADACP1", "AAMDC", "AASS", "ABAT", "AASDH")
collection <- list(real = mySig, fake = mySig2)
# now test for enrichment using ROAST
GSEAres <- mroast(normData, collection, design, contrast.matrix[,1], nrot=99) 
GSEAres

# run GSEA using GSVA package----
#be aware that if you choose a large MsigDB file here, this step may take a while
#first, convert your geneSetCollections into lists using the 'geneIds' function
broadSet.C7 <- geneIds(broadSet.C7)
GSVA.res.c7 <- gsva(normData$E, #your data
                    broadSet.C7, # your gene set collection
                    min.sz=5, max.sz=500, #criteria for filtering gene sets
                    verbose=FALSE,
                    mx.diff=FALSE,
                    rnaseq=TRUE,
                    method="gsva") #options for method are "gsva", ssgsea', "zscore" or "plage"
#extract the enrichment scores from this result
GSVA.res.c7 <- GSVA.res.c7$es.obs

# Apply linear model to GSVA result ----
# now using Limma to find significantly enriched gene sets in the same way you did to find diffGenes
# this means you'll be using topTable, decideTests, etc
# note that you need to reference your design and contrast matrix here
fit.C7 <- lmFit(GSVA.res.c7, design)
ebFit.C7 <- eBayes(fit.C7)

# use topTable and decideTests functions to identify the differentially enriched gene sets
topPaths.C7 <- topTable(ebFit.C7, adjust ="BH", coef=1, number=50, sort.by="logFC")
res.C7 <- decideTests(ebFit.C7, method="global", adjust.method="BH", p.value=0.05, lfc=0.58)
#the summary of the decideTests result shows how many sets were enriched in induced and repressed genes in all sample types

# extract the enrichment matrix from the GSVA result and convert to an expressionSet object 
head(res.C7)
myEset.C7 <- new("ExpressionSet", exprs = GSVA.res.c7)
head(myEset.C7)

# pull out the gene sets that are differentially enriched between groups ----
diffSets.C7 <- myEset.C7[res.C7[,1] !=0 | res.C7[,2] !=0 | res.C7[,3] !=0]
diffSets.C7 <- exprs(diffSets.C7)
head(diffSets.C7)
dim(diffSets.C7)

# make a heatmap of differentially enriched gene sets ----
hr.C7 <- hclust(as.dist(1-cor(t(diffSets.C7), method="pearson")), method="complete") #cluster rows by pearson correlation
hc.C7 <- hclust(as.dist(1-cor(diffSets.C7, method="spearman")), method="complete") #cluster columns by spearman correlation

# Cut the resulting tree and create color vector for clusters.  Vary the cut height to give more or fewer clusters, or you the 'k' argument to force n number of clusters
library(RColorBrewer)
mycl.C7 <- cutree(hr.C7, k=2)
mycolhc.C7 <- rainbow(length(unique(mycl.C7)), start=0.1, end=0.9) 
mycolhc.C7 <- mycolhc.C7[as.vector(mycl.C7)] 

#load the gplots package for plotting the heatmap
library(gplots) 
#assign your favorite heatmap color scheme. Some useful examples: colorpanel(40, "darkblue", "yellow", "white"); heat.colors(75); cm.colors(75); rainbow(75); redgreen(75); library(RColorBrewer); rev(brewer.pal(9,"Blues")[-1]). Type demo.col(20) to see more color schemes.
myheatcol <- greenred(75)
#plot the hclust results as a heatmap
heatmap.2(diffSets.C7, 
          Rowv=as.dendrogram(hr.C7), Colv=as.dendrogram(hc.C7), 
          col=myheatcol, scale="row",
          labRow = NA,
          density.info="none", trace="none", 
          cexRow=0.9, cexCol=1, margins=c(10,25)) # Creates heatmap for entire data set where the obtained clusters are indicated in the color bar.

#print your enrichment results to an excel spreadsheet
write.table(diffSets.C7, "diffSets_C7.xls", sep="\t", quote=FALSE)

# OPTIONAL: Calculate the % overlap between gene sets ----
library(lattice)
SigOverlap <- computeGeneSetsOverlap(broadSet.C7, rownames(normData$E))
rgb.palette <- colorRampPalette(c("blue", "red", "yellow"), space = "rgb")
levelplot(SigOverlap, main="geneset correlation matrix", xlab="", ylab="", col.regions=rgb.palette(1000), cuts=100, cexRow=0.5, cexCol=1, margins=c(10,10))

# OPTIONAL: make custom gene set collections----
#depending on how many genesets are enriched, you may want to filter your enrichement results
#here, I filter the C2 enrichment results to get only KEGG, BIOCARTA, REACTOME
#easiest way to do this is to just open the enrichment results and manually look at what rows correspond to the different subsets of interest
diffSets.KEGG <- diffSets.B6[1:8,]
setNames.KEGG <- rownames(diffSets.KEGG)

diffSets.BIOCARTA <- diffSets.B6[9:29,]
setNames.BIOCARTA <- rownames(diffSets.BIOCARTA)

diffSets.PID <- diffSets.B6[34:36,]
setNames.PID <- rownames(diffSets.PID)

diffSets.REACTOME <- diffSets.B6[37:66,]
setNames.REACTOME <- rownames(diffSets.REACTOME)

#or you can import your own edited list containing a subset of enriched pathways
diffSets.custom <- as.matrix(read.delim("diffSets_C2_custom.txt", header=TRUE, row.names=1))
setNames.custom <- rownames(diffSets.custom)
