# Introduction to this script----
# This script accomplishes several goals:
# 1st - it handles importing your kallisto quantification data into R
# 2nd - this script introduces us to basic tools for annotation
# 3rd - it uses Sleuth to identify differentially expressed genes or transcripts directly from the Kallisto output
# A minimal representation of this script is contained in the last code chunk called 'the essentials'.

# load packages ----
library(tidyverse) # provides access to Hadley Wickham's collection of R packages for data science, which we will use throughout the course
library(tximport) # package for getting Kallisto results into R
library(ensembldb) #helps deal with ensembl
library(EnsDb.Hsapiens.v79) #replace with your organism-specific database package
library(biomaRt) # an alternative for annotation
library(beepr) #just for fun

# read in your study design ----
#there are LOTS of ways to read data into R, but the readr package (from tidyverse) is one of the simplest
targets <- read_tsv("../../studyDesign.txt")

# you can easily create file paths to the abundance files generated by Kallisto using the 'file.path' function
path <- file.path("../readMapping", targets$sample, "abundance.h5")
# now check to make sure this path is correct by seeing if the files exist
all(file.exists(path)) 
# this should look familiar from your homework
# the functions which(), any() and all() take logical vectors as their argument. The any()
# function will return TRUE if one or more of the elements in the logical vector is TRUE. The all() function
# will return TRUE if every element in the logical vector is TRUE.

# use dplyr to modify your study design to include these file paths as a new column.  
# Sleuth looks for this column, which must be called 'path', in order to locate the data output from Kallisto
targets <- mutate(targets, path)

# get annotations using organism-specific package ----
# map transcript identifiers to gene symbols ----
Tx <- transcripts(EnsDb.Hsapiens.v79, columns=c(listColumns(EnsDb.Hsapiens.v79, "tx"), "gene_name"))
Tx <- as.data.frame(Tx)
head(Tx)

#need to change first column name to 'target_id'
Tx <- dplyr::rename(Tx, target_id = tx_id)
row.names(Tx) <- NULL
head(Tx)
#transcrip ID needs to be the first column in the dataframe
Tx <- Tx[,c(6,11)]
#write.table(Tx,"annotation_transcripts.xls", sep="\t", quote=FALSE)
head(Tx)

# get annotations using BiomaRt----
listMarts() #default host is ensembl.org, and most current release of mammalian genomes
#listMarts(host="parasite.wormbase.org") #access to parasite worm genomes
#listMarts(host="protists.ensembl.org") #access to protozoan genomes

#choose the 'mart' you want to work with
myMart <- useMart(biomart="ENSEMBL_MART_ENSEMBL")
#take a look at all available datasets within the selected mart
available.datasets <- listDatasets(myMart)
#now grab the ensembl annotations for human
Hs.anno <- useMart(biomart="ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl")
Hs.filters <- listFilters(Hs.anno)

Tx <- getBM(attributes=c('ensembl_transcript_id_version',
                                     'external_gene_name'),
            mart = Hs.anno)

Tx <- as_tibble(Tx)
#we need to rename the two columns we just retreived from biomart
Tx <- dplyr::rename(Tx, target_id = ensembl_transcript_id_version, gene_name = external_gene_name)


# import Kallisto transcript counts into R using Tximport ----
# copy the abundance files to the working directory and rename so that each sample has a unique name
Txi_gene <- tximport(path, 
                     type = "kallisto", 
                     tx2gene = Tx, 
                     txOut = FALSE, #How does the result change if this =FALSE vs =TRUE?
                     countsFromAbundance = "lengthScaledTPM",
                     ignoreTxVersion = TRUE)

#take a look at the type of object you just created
class(Txi_gene)
names(Txi_gene)

myCPM <- as_tibble(Txi_gene$abundance, rownames = "geneSymbol") # these are you counts after adjusting for transcript length
myCounts <- as_tibble(Txi_gene$counts, rownames = "geneSymbol") # these are your transcript per million (TPM) values, or counts per million (CPM) if you collapsed data to gene level

# #if you exported transcript level data and want to append your gene symbols to the data frame
# Txi_trans <- as_tibble(Txi_trans$counts, rownames = "target_id")
# Txi_trans <- left_join(Txi_trans, Tx)

# the essentials ----
# this chunk contains the minimal essential code from this script
library(tidyverse) # provides access to Hadley Wickham's collection of R packages for data science, which we will use throughout the course
library(tximport) # package for getting Kallisto results into R
library(biomaRt) # provides access to a wealth of annotation info
targets <- read_tsv("../../studyDesign.txt")# read in your study design
path <- file.path("../readMapping", targets$sample, "abundance.h5") # set file paths to your mapped data
targets <- mutate(targets, path) # add paths to your study design (only necessary for Sleuth)
Hs.anno <- useMart(biomart="ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl") # select 'mart' from biomaRt for annotations
Tx <- getBM(attributes=c('ensembl_transcript_id_version', # get gene symbols for each transcript ID
                         'external_gene_name'),
            mart = Hs.anno)
Tx <- as_tibble(Tx) # convert this annotation mapping file to a tibble (the tidyverse version of a dataframe)
Txi_gene <- tximport(path, #reading kallisto data into R
                     type = "kallisto", 
                     tx2gene = Tx, 
                     txOut = FALSE, 
                     countsFromAbundance = "lengthScaledTPM")
myCPM <- as_tibble(Txi_gene$abundance, rownames = "geneSymbol") # these are you counts after adjusting for transcript length
myCounts <- as_tibble(Txi_gene$counts, rownames = "geneSymbol") # these are your transcript per million (TPM) values, or counts per million (CPM) if you collapsed data to gene level
