# Introduction to this script----
# In this script you'll read your transcript abundance counts into R
# To do this, you'll use the 'tximport' package which is a new and relatively flexible R package designed
# specifially for handling transcript counts generated from a number of programs (Kallisto, Sailfish, Salmon, etc).

# Load packages ----
library(dplyr)
library(readr) #general use package from Hadley Wickham for reading in tables
library(Biostrings) #we'll use this to edit our reference fasta file
library(tximport) #package for getting Kallisto results into R
library(ensembldb) ##used together with your organism-specific database package to get annotation info
library(EnsDb.Hsapiens.v86) #replace with your organism-specific database package

# OPTIONAL: modify reference transcriptome .fasta file ----
# depending on your reference transcriptome, we may need to edit the fasta file
# Transcript identifiers should NOT have a version number appended to the ID
# this creats a problem later when we try to append annotations based on these IDs
# you would only need to edit this file once
Hs.trans <- readDNAStringSet("Homo_sapiens.GRCh38.cdna.all.fa")
slotNames(Hs.trans) #look at the 'slots' in the object you just created
Hs.trans@ranges@NAMES #access one of these slots
#replace transcript names with new names that don't have decimal version number
names(Hs.trans) <- gsub("(\\.).*", "", names(Hs.trans))
writeXStringSet(Hs.trans, "Hs.trans.shortnames.fasta")

# align reads using Kallisto ----
# this is a unix based program, so we'll access it using the 'system' function
# note: this will only work if you have installed Kallisto on your computer
system("kallisto")
system("ls -l")
system("chmod u+x ./readMapping.sh")
system("./readMapping.sh")


# Get annotations ----
# If we want to know what kinds of data are retriveable, 
# use EnsemblDB package functions to look at the tables of the annotation database
listTables(EnsDb.Hsapiens.v86)
# now take a look at the columns of the "tx" table within the database
listColumns(EnsDb.Hsapiens.v86, "tx")

#use the 'transcripts' function from the EnsemblDB package to get annotation info
Tx <- transcripts(EnsDb.Hsapiens.v86, 
                  columns=c(listColumns(EnsDb.Hsapiens.v86,
                                        "tx"), "gene_name"))

Tx <- as.data.frame(Tx)
head(Tx)

#need to change first column name to 'target_id'
Tx <- dplyr::rename(Tx, target_id = tx_id)
row.names(Tx) <- NULL
head(Tx)

#transcript ID needs to be the first column in the dataframe
Tx <- Tx[,c(6,12)]

# OPTIONAL: Get annotations for other organisms using BiomaRt----
library(biomaRt)
listMarts() #default host is ensembl.org, and most current release of mammalian genomes
#listMarts(host="parasite.wormbase.org") #access to parasite worm genomes
#listMarts(host="protists.ensembl.org") #access to protozoan genomes

#choose the 'mart' you want to work with
myMart <- useMart(biomart="ENSEMBL_MART_ENSEMBL")
#take a look at all available datasets within the selected mart
available.datasets <- listDatasets(myMart)
#I'll use canine as an example
canFam <- useMart(biomart="ENSEMBL_MART_ENSEMBL", dataset = "cfamiliaris_gene_ensembl")
canFam.filters <- listFilters(canFam)

TxIDs_to_genes <- getBM(attributes=c('ensembl_transcript_id_version',
                                     #'ensembl_transcript_id',
                                     'external_gene_name'),
                        mart = canFam)



# Import Kallisto transcript counts into R using Tximport ----
# copy the abundance files to the working directory and rename so that each sample has a unique name
# now, capture the names of your abundance files generated by Kallisto
targets <- read.table("Crypto_studyDesign.txt", row.names=NULL, header = T, as.is = T)
files <- file.path(targets$sample, "abundance.h5")
# now check to make sure this path is correct by seeing if the files exist
all(file.exists(files))

Txi_gene <- tximport(files, 
                     type = "kallisto", 
                     tx2gene = Tx, 
                     txOut = TRUE, #How does the result change if this =FALSE vs =TRUE?
                     countsFromAbundance = "lengthScaledTPM")

#take a look at the object you just created
head(Txi_gene$counts)
head(Txi_gene$abundance)

#QUIZ----
#1. what class are these objects, and what are their dimensions?
#2. what should the columns of Txi_gene$abundance sum to, what about Txi_gene$counts?
#3. how would you quickly find out the answer to #2 in R?