---
title: 'Analysis of scRNA-seq data using R'
subtitle: 'Lecture 14 •  watch by December 8, 2021'
date: 2018-12-28 00:00:00
description: In this final lecture of the course, you'll learn to use Seurat to analyze scRNA-seq data, including carrying out dimensional reduction and display using UMAP, identifying cell clusters and cluster-specific marker genes, and how to integrate data from multiple samples.
featured_image: '/images/drops.png'
---

No lecture slides for this class.  We'll spend the entire time working on scRNA-seq script.

## Overview

With your data already preprocessed with Kallisto-Bustools, you're now ready to import into R and use a variety of packages to filter, plot and analyze your data.

## Learning objectives

* Be able to import preprocessed data into R and create a Seurat object
* Carry out filtering using [DropUtils](https://bioconductor.org/packages/release/bioc/html/DropletUtils.html)
* Generate a Quality Control report of your scRNA-seq data directly within R
* Use standard QC metrics and plots to filter your data
* Generate clusters and visualize via UMAP dimensional reduction
* Find cluster-specific marker genes with Seurat
* Annotate unknown clusters using public databases and [CellAssign](https://www.rdocumentation.org/packages/cellassign/) and [SingleR](https://bioconductor.org/packages/release/bioc/html/SingleR.html)
* Integrate multiple samples and use sample details to analyze integrated data

## Code and files

**[pre-processed data for 1000 PBMCs](https://drive.google.com/drive/folders/1RO45z5DEVpuaq5qwlF5QNdhc0tbGVK7l?usp=sharing)** - You only need to download this if you were unable to use kb-python in the last lecture to process raw scRNA-seq data.  This ensures that everyone can follow along with this lecture, regardless of whether you were able to install or use Kb-python.

**[DIY_scRNAseq.R](http://DIYtranscriptomics.github.io/Code/files/DIY_scRNAseq.R)** - this is the R script that we'll use for this lecture.

**[functions.R](http://DIYtranscriptomics.github.io/Code/files/functions.R)** - this is the custom R function we'll use for generating a QC report with our scRNA-seq data (see Reading material below for source).

**[Seurat objects](https://drive.google.com/open?id=1SEEr70W6D9itvVaLfEXuRVWqyj_Gribc&authuser=danielbeiting%40gmail.com&usp=drive_fs)** - this folder contains two Seurat objects from an unpublished mouse experiment (courtesy of Chris Hunter's lab).  One sample is from a naive control mouse, while the second is from a mouse infected with the protozoan parasite, *Toxoplasma gondii*.  We'll use these data in the second 1/2 of the lecture to practice integration and differential gene testing between conditions.

## Lecture videos

### Part 1 – Importing scRNA-seq data into R and carrying out basic QA analysis.

<iframe src="https://player.vimeo.com/video/650710835" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>

### Part 2 – Dimensional reduction with UMAP, and cluster identification

<iframe src="https://player.vimeo.com/video/650714817" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>

### Part 2 – Integration of multiple samples and working with sample metadata

<iframe src="https://player.vimeo.com/video/650714817" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>

---

## Reading

[EmptyDrops: distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data.](https://10.1186/s13059-019-1662-y). - This is the paper describing the DropletUtils package that we use in this lecture to identify empty drops.

[Sarah Ennis' Github repo for preprocessing scRNA-seq data](https://github.com/Sarah145/scRNA_pre_process) - This is the source of the custom script we use to generate the CellRanger-esque html QC report. 

[Probabilistic cell-type assignment of single-cell RNA-seq for tumor microenvironment profiling](https://pubmed.ncbi.nlm.nih.gov/31501550/) - Describes the CellAssign algorithm and R package that we use to identify clusters.

[Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage](https://doi.org/10.1038/s41590-018-0276-y) – describes the SingleR and celldex packages that allow us to leverage bulk RNA-seq data in public repositories to curate clusters in our scRNA-seq.

[Comprehensive Integration of Single-Cell Data](https://doi.org/10.1016/j.cell.2019.05.031) - This 2019 paper describes the underlying 



---

## Other videos

### Letitia Parcalabescu (and Ms. Coffee Bean) explains UMAP in 10min.  Great video!

<iframe width="560" height="315" src="https://www.youtube.com/embed/6BPl81wGGP8" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

---

## Acknowledgements

I owe a big THANK YOU to Lindsey Shallberg and Dr. Chris Hunter for their willingness to share unpublished scRNA-seq data from the spleen of Toxoplasma-infected mice, which is used at the end of this lecture to illustrate data integration and downstream analysis.

