---
title: 'Making your analysis portable and reproducible with Docker'
subtitle: 'Lecture 15 • 2019.12.11 • Levin 111 (SAIL classroom) • 3-5pm'
date: 2018-12-29 00:00:00
description: Reproducing an analysis requires more than just code. You need the original raw data, access to the appropriate programming languages, and application specific packages (and often specific versions of these packages).  This poses a major impediment to reproducibility, even for researchers with a background in bioinformatics.  To address this challenge, you'll learn how to 'containerize'  your data, scripts and software into a code timecapsule, making it easy to share and rerun an entire analysis with the push of a button. 
featured_image: '/images/capsule.jpg'
---

<div style="text-align: right"> Image credit: 'Emoh: sculpture / time capsule & temporary home' by Loren Naji </div>

[Lecture slides](https://www.icloud.com/keynote/0Nadab6WL2IRMYPcZn8zAAcww#Lecture13%5Freproducibility)

## Overview

Reproducing an analysis requires more than just code. You need the original raw data, access to the appropriate programming languages, and application specific packages (and often specific versions of these packages).  This poses a major impediment to reproducibility, even for researchers with a background in bioinformatics.  To address this challenge, you'll learn how to 'containerize'  your data, scripts and software into a code timecapsule, making it easy to share and rerun an entire analysis with the push of a button.  

## Goals

* Review the answer for hackdash #2
* Brief review of Rmarkdown
* Create a reproducible R environment using the [renv package](https://rstudio.github.io/renv/articles/renv.html)
* Discuss the basics of Docker and containerized software
* Create your own CodeOcean account using your course email (comes with 15 compute hrs/month, free)
* Interact with our 'course in a capsule'

## Code

Clicking on the embedded code capsule below will take you directly to Code Ocean.  Simply login (or set-up a free account if you don't already have one) and you'll be able to re-run the entire course in a matter of minutes, without any software installation or data download.  Your first run may take ~15min, since the full computing environment must be being built, but subsequent runs will be much faster.  Note that this capsule includes raw fastq files, kallisto outputs, all of MSigDB for running GSEA, and the entire ARCHS4 database for interrogating ~700,000 publically available mouse and human RNAseq datasets.  Have fun adapting this capsule for your own analyses!

<script src="https://codeocean.com/widget.js?slug=1031026" async></script>

## Reading

[Whitepaper](http://DIYtranscriptomics.github.io/Reading/files/codeOcean_whitepaper.pdf) describing CodeOcean, the first cloud-based computational reproducibility platform.

CodeOcean is based on Docker, a free and open-source tool that allows you to 'build, share and run applications anywhere'.  Take a look at this [intro to Docker](https://docker-curriculum.com/) to get learn more.

[On-boarding document](https://docs.google.com/document/d/1baA8IRFDZE1rJNB3SYk7Aksvuz5AJ9W4HG2iqYoPt-8/edit?usp=sharing) for gettign started with Code Ocean

[recent paper](https://stm.sciencemag.org/content/11/519/eaax4204) from my lab, showing a code capsule embedded directly in the joural webpage. 



