---
title: 'Analysis of scRNA-seq data using R'
subtitle: 'Lecture 14 •  watch by December 8, 2021'
date: 2018-12-28 00:00:00
description: In this final lecture of the course, you'll learn to use Seurat to analyze scRNA-seq data, including carrying out dimensional reduction and display using UMAP, identifying cell clusters and cluster-specific marker genes, and how to integrate data from multiple samples.
featured_image: '/images/umap.png'
---

## Overview

With your data already preprocessed with Kallisto-Bustools, you're now ready to import into R and use a variety of packages to filter, plot and analyze your data.

## Learning objectives

* Be able to import preprocessed data into R and create a Seurat object
* Carry out filtering using [DropUtils](https://bioconductor.org/packages/release/bioc/html/DropletUtils.html)
* Generate a Quality Control report of your scRNA-seq data directly within R
* Use standard QC metrics and plots to filter your data
* Generate clusters and visualize via UMAP dimensional reduction
* Find cluster-specific marker genes with Seurat
* Annotate unknown clusters using public databases and [CellAssign](https://www.rdocumentation.org/packages/cellassign/) and [SingleR](https://bioconductor.org/packages/release/bioc/html/SingleR.html)
* Integrate multiple samples and use sample details to analyze integrated data

## Code and files

If you were unable to use kb-python in the last lecture to process raw scRNA-seq data, you'll need to download pre-processed data [here](https://drive.google.com/drive/folders/1RO45z5DEVpuaq5qwlF5QNdhc0tbGVK7l?usp=sharing), which will allow you to follow along with this lecture.

[DIY_scRNAseq.R](http://DIYtranscriptomics.github.io/Code/files/DIY_scRNAseq.R) - this is the main R script that we'll use for the remainder of the course.

[functions.R](http://DIYtranscriptomics.github.io/Code/files/functions.R) - this is the custom function we'll use for generating a QC report with our scRNA-seq data (see Reading material below for source).

## Lecture videos

Coming soon!

---

## Reading

[EmptyDrops: distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data.](https://10.1186/s13059-019-1662-y). - This is the paper describing the DropletUtils package that we use in this lecture to identify empty drops.

[Sarah Ennis' Github repo for preprocessing scRNA-seq data](https://github.com/Sarah145/scRNA_pre_process) - This is the source of the custom script we use to generate the CellRanger-esque html QC report. 

[Probabilistic cell-type assignment of single-cell RNA-seq for tumor microenvironment profiling](https://pubmed.ncbi.nlm.nih.gov/31501550/) - Describes the CellAssign algorithm and R package that we use to identify clusters.

[Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage](https://doi.org/10.1038/s41590-018-0276-y) – describes the SingleR package that allows us to use bulk RNA-seq data in public repositories to curate clusters in our scRNA-seq.


---

## Other videos

### Letitia Parcalabescu (and Ms. Coffee Bean) explains UMAP in 10min.  Great video!

<iframe width="560" height="315" src="https://www.youtube.com/embed/6BPl81wGGP8" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

