---
title: 'Single cell RNA-seq – principles and processing'
subtitle: 'Lecture 13 •  watch by December 1, 2021'
date: 2018-12-29 00:00:00
description: Now that you're comfortable with bulk RNA-seq data analysis, we'll shift our focus to the rapidly developing landscape of single cell RNA-seq (scRNA-seq).  In this lecture, you'll learn about the underlying technology and we'll demonstrate how to process raw single cell data directly on your laptop (!) for importing into R/bioconductor.
featured_image: '/images/colorbins.png'
---

## Overview

Now that you're comfortable with bulk RNA-seq data analysis, we'll shift our focus to the rapidly developing landscape of single cell RNA-seq (scRNA-seq).  In this lecture, you'll learn about the underlying technology and we'll demonstrate how to process raw single cell data directly on your laptop (!) for importing into R/bioconductor.

## Learning objectives

* Understand droplet-based scRNA-seq technology
* Be able to compare and contrast single cell and bulk RNA-seq methods 
* Understand cost and experimental design considerations for scRNA-seq experiments.
* Familiarity with multiplexed single cell assays (CITE-seq, 'multiome', TEA-seq)
* Be able to define common terms and concepts in single cell genomics
* Use Kallisto-BUStools to preprocess raw scRNA-seq data (via [kb-python](https://www.kallistobus.tools/))


## What you need to do

* **[Download raw files](https://drive.google.com/drive/folders/1DbLRO4kv-y3W06adFR26RdSaDPmfB4UA?usp=sharing)**.  You will need about **5Gb** of storage space on your harddrive to accomodate this download.  *please do not uncompress these files (leave them as .gz files)*.  This is data from 1000 peripheral blood mononuclear cells (PBMCs) and is one of the sample datasets provided by 10X Genomics [here](https://bit.ly/10xPBMC_small).  I have merged the separate lane files to make this simpler to work with for the course.  

* You'll need the human transcriptome index file that you created way back in [lecture 2](https://diytranscriptomics.com/project/lecture-02).  If you don't have this, remember it's easy to create using:

```bash
# activate your rnaseq environment and create the index
conda activate rnaseq
kallisto index -i Homo_sapiens.GRCh38.cdna.all.index Homo_sapiens.GRCh38.cdna.all.fasta
```

* You will need to have kb-python installed in a Conda environment on your laptop.  We did this way back in [lecture 1](https://diytranscriptomics.com/project/lecture-01).  If you can't get this software installed, no worries, just download the already processed data [here](), and focus on the concepts in the lecture.

## Lecture video

Coming soon!

---

## Reading

Páll Melsted, A. Sina Booeshaghi, Fan Gao, Eduardo Beltrame, Lambda Lu, Kristján Eldjárn Hjorleifsson, Jase Gehring, Lior Pachter. (2019). Modular and efficient pre-processing of single-cell RNA-seq. bioRxiv; doi: https://doi.org/10.1101/673285

Páll Melsted, Vasilis Ntranos, Lior Pachter. (2019). The barcode, UMI, set format and BUStools, Bioinformatics; doi: https://doi.org/10.1093/bioinformatics/btz279

* In addition to paired-end raw fastq files, the folder you downloaded above also contains a ['barcode whitelist'](https://kb.10xgenomics.com/hc/en-us/articles/360031133451-Why-is-there-a-discrepancy-in-the-3M-february-2018-txt-barcode-whitelist-) (3M-february-2018_v3.txt) that contains all the barcodes for the 10X RNA-seq assay.



---


