---
title: 'Wrangling gene expression data'
subtitle: 'Lecture 6 • 2019.02.27 • Hill 130'
date: 2019-01-07 00:00:00
description: We'll use the Tidyverse tools to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data quickly and easily. 
featured_image: '/images/tidyDNA.jpg'
---

[Lecture slides](https://www.icloud.com/keynote/0i4VPYHj9N0286jLLF20BYipw#Lecture05%5FexploratoryAnalysis)

[Homework: Introduction to the Tidyverse (~4hrs)](https://www.datacamp.com/courses/introduction-to-the-tidyverse) - due before the start of class on March 6th.


## Overview

In this class we'll use Hadley Wickham's [dplyr](http://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html) package along with other tools from his [tidyverse](https://www.tidyverse.org/) of packages to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data sets quickly and easily using simple commands in R.

## Goals

* Start/finish step 2 script
* Begin to explore our data in R using ggplot2
* Produce interactive graphic with plotly
* Discuss basics of reshape package
* Talk about color palettes in R
* Filter data to remove lowly expressed genes
* Normalize data

## Code

[Step 2 script](http://DIYtranscriptomics.github.io/Code/files/Step2_dataExploration.R)

## Reading

Hadley Wickham (author of Dplyr, Reshape2 and ggplot2 packages) has a nice pre-print paper on the key aspects of making ['Tidy Data'](http://vita.had.co.nz/papers/tidy-data.pdf)
