---
title: 'Course overview and intro to RNAseq technology and data'
subtitle: 'Lecture 1 • precourse material'
date: 2019-01-12 00:00:00
description: After a brief overview of the course, we'll spend the opening lecture talking about Illumina's 'Sequencing by Synthesis' technology, and walk through the steps involved in planning for a transcriptomics experiment. 
featured_image: '/images/illumina_flowcell.jpg'
---

[Course overview slides](https://www.icloud.com/keynote/0dABsRGX-LxA3pHuwCUgejkhg#precourse%5Fintro)

[Intro to RNA-seq lecture slides on iCloud](https://www.icloud.com/keynote/0n2tSKAVc2chal9Wt8cj-uDiQ#Lecture01%5FRNAseqIntro)

[Homework #1: DataCamp Intro to R course (~2hrs)](https://www.datacamp.com/courses/free-introduction-to-r) - due 3pm next Wednesday, April 22nd.

After a brief overview of the course, we'll spend the opening lecture talking about Illumina's 'Sequencing by Synthesis' technology, and walk through the steps involved in planning for a transcriptomics experiment.  After this lecture, jargon like flow cell, single-end, paired-end, indexing, cluster density, library, ribo-depletion, scRNAseq, and fastq will practically roll off your tongue.

Although there are no formal prerequisties for the course and no official course text book, there are a few important housekeeping things you'll need to take care of before the first class.  **Scroll down to the bottom of the page and watch the introductory video.**

## Please complete before the first class

If you officially registered for the course with Dr. Beiting, then you have access to DataCamp for course homework and to our course Slack workspace for communicating with other students and your TAs.  If you did not register for the course, you will not have access to DataCamp or the Slack workspace, but will still be able to follow the video lectures for each module.

**[Get access to course DataCamp portal](https://www.datacamp.com/)**.  You will receive an email from us with a link to access the course portal on the datacamp site. If you already have a DataCamp account under your UPenn email, then this account will be granted access to the course portal.  DataCamp is a online learning environment that is used throughout the course for homework assignments.  This is a subscription service that is being provided to you free-of-charge (courtesy of DataCamp) only during the current course, so take full advantage to explore any other learning materials on the site. 

[Download and sign-up for Slack](https://slack.com/), a messaging platform we will use heavily during the course.  *All course communication will happen through Slack, not email*.  Questions related to material covered in a lecture should be posted to the specific Slack channel associated with that lecture.  Questions will be reviewed by the TAs and used as guidance for TA help sessions.  The Slack page comes equipped with standard messaging capabilities, but also includes video and voice calling features, as well as screensharing, making it easy for you to connect with your classmates and instructors when you have problems or questions. 

[Complete this skills survey](https://bit.ly/DIYtranscriptomics_start). 


## Other useful info

The course website is meant to be the go-to place for all content related to the course, including lecture slides, recordings, reading material, code and more.  Below are some additional resources that I will refer to throughout class.

[Protocols website](https://protocols.hostmicrobe.org) - This is where my lab posts all protocols/SOPs for RNAseq library preparation, as well as bioinformatics workflows.

## Welcome video

Please watch the video below for a introduction to the course, how you will interact with the course materials on this website, and how you can communicate with your classmates and instructors.

<iframe src="https://player.vimeo.com/video/405054924" width="640" height="338" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>

---

## Learning objectives

* Get familiar with jargon of RNAseq/HTS (paired-end, cluster density, fastq, index, etc)
* Understand Illumina’s “Sequencing by Synthesis” (SBS) technology
* Set priorities when planning a sequencing experiment
* Understand the basics of library prep
* Have a sense for how single cell sequencing (scRNAseq) and spatial transcriptomics fit into the evolving landscape of transcriptomics

---

## Reading

[File formats produced by sequencing](http://binf.snipcademy.com/lessons/sequence-file-formats).

[RNA-seqlopedia](http://rnaseq.uoregon.edu/) - Created by the Univ. of Orgeon, this is a great resource for understanding the entire RNAseq workflow.

---

## Lecture video

### Part 1 - History of sequencing and details of Illumina's 'Sequencing by Synthesis' (SBS) technology

<iframe src="https://player.vimeo.com/video/408044306" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>

### Part 2 - Design considerations for RNAseq experiments, and the state-of-the-art for transcriptomics

<iframe src="https://player.vimeo.com/video/408069325" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>

---