---
title: 'Multivariate analysis'
subtitle: 'Lecture 7 • 2019.03.06 • Hill 130 • 3-5pm'
date: 2019-01-06 00:00:00
description: In this class we’ll use multivariate statisical methods, including Principal Component Analysis (PCA), to explore how experimenal covariates contribute to variance in our data.  We’ll discuss common missteps and sources of variance in transcriptional data sets.  You’ll also continue to build your plotting skills by using ggplot2 to graph the results of PCA analyses.
featured_image: '/images/dataExplore.jpg'
---

[Lecture slides](https://www.icloud.com/keynote/0i4VPYHj9N0286jLLF20BYipw#Lecture05%5FexploratoryAnalysis)

[Homework: Introduction to the Tidyverse (~4hrs)](https://www.datacamp.com/courses/introduction-to-the-tidyverse) - due before the start of class today!

## Overview

In this class we’ll use multivariate statisical methods, including Principal Component Analysis (PCA), to explore how experimenal covariates (e.g., sex, age, treatment) contribute to the overall structure of our data.  We’ll discuss common missteps and sources of variance in transcriptional data sets.  You’ll also continue to build your plotting skills by using ggplot2 to graph the results of PCA analyses.

## Goals

* Start/finish step 3 script
* Discuss basics of multivariate statistical analysis
* Carry out hierarchical clustering of samples
* Discuss and perform principal component analyses (PCA)
* Produce 'small multiples' plot
* Use standard dplyr 'verbs' to quickly query our data
* Produce interactive graphics using the plotly package
* Produce interactive tables with DT package

## Code

[Step 3 script](http://DIYtranscriptomics.github.io/Code/files/Step3_dataWrangling.R)



## Reading

I mentioned various unsupervised methods for dimensional reduction of your data (PCA, MDS, T-SNE).  In particular, T-SNE has become popular in representing single-cell RNAseq data, but it is also one of the more complex visualization methods to understand.  Although we didn't discuss this in class, I wanted to include a link to a [great blog post describing T-SNE](http://distill.pub/2016/misread-tsne/), as well as [the original T-SNE paper](http://DIYtranscriptomics.github.io/Reading/files/TSNE.pdf).  Please familiarize yourself with these if you plan on using this visualziation method.
 