---
title: 'Wrangling gene expression data'
subtitle: 'Lecture 6 • Friday • 2020.05.01 • 3-5pm'
date: 2019-01-07 00:00:00
description: We'll begin this class by filtering and normalzing our data, all while using the ggplot2 graphing package to visualize the impact these changes have our data. You'll also be introduced to Hadley Wickham's philosophy of 'tidy data' by using the dplyr package, expanding your understanding of tools within the Tidyverse.  
featured_image: '/images/tidyDNA.jpg'
---

<div style="text-align: right"> Image credit: Daniel Horowitz for NPR </div>

[Lecture slides](https://www.icloud.com/keynote/0wOhMvs1a4AZp-v4OMau1j67A#lecture05%5Frefslides) -  Although the entire class was spent working on the Step 2 script, I briefly referenced a few slides, which I've included here.

[Homework: Data visualization with ggplot2 - part 1 (~4hrs)](https://www.datacamp.com/courses/data-visualization-with-ggplot2-1) - due before the start of class on Oct 16th.


## Overview

We'll begin this class by filtering and normalzing our data, all while using the ggplot2 graphing package to visualize the impact these changes have our data. You'll also be introduced to Hadley Wickham's philosophy of 'tidy data' by using the [dplyr](http://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html) package, expanding your understanding of tools within the [tidyverse](https://www.tidyverse.org/). 

## Goals

* Start/finish step 2 script
* Begin to explore our data in R using ggplot2
* Discuss basics of Tidy data and the reshape2 package
* Talk briefly about color palettes in R
* Filter data to remove lowly expressed genes
* Normalize data

## Code

[Step 2 script](http://DIYtranscriptomics.github.io/Code/files/Step2_dataWrangling.R)


## Reading

[Tidy Data](http://vita.had.co.nz/papers/tidy-data.pdf) - Hadley Wickham (author of Tidyverse packages and Chief Scientist at RStudio) describes the philosophy of tidy data in this paper. 

[Grammar of graphics](http://DIYtranscriptomics.github.io/Reading/files/layered-grammar.pdf) - Another paper by Hadley Wickham.  This one explains the rationale behind ggplot2.

[ggplot2 cheatsheet](http://DIYtranscriptomics.github.io/Reading/files/ggplot2-cheatsheet.pdf) – a very helpful guide as we continue to use ggplot2 for all of our plotting needs.

[catalog of R graphs](http://shiny.stat.ubc.ca/r-graph-catalog/) - Take a look at some of the various ways to graph your data and the underlying R code in this

[R Graphics Cookbook](http://hdl.library.upenn.edu/1017.12/1675994) - If you end up using R to make a lot of graphs, you will find the to be an important reference. It's available free to UPenn folks as an Ebook.

[original TMM normalization manuscript](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25).


---

## Lecture video

Will be posted prior to this class time slot

---