---
title: 'Wrangling gene expression data'
subtitle: 'Lecture 6 • 2019.02.27 • Hill 130 • 3-5pm'
date: 2019-01-07 00:00:00
description: After filtering and normalizing our data in R, we'll apply the 'grammar of graphics' and the principles of 'tidy data' to plot, change, sort, filter, arrange and summarize expression data quickly and easily. 
featured_image: '/images/tidyDNA.jpg'
---

<div style="text-align: right"> Image credit: Daniel Horowitz for NPR </div>

[Lecture slides](https://www.icloud.com/keynote/0wOhMvs1a4AZp-v4OMau1j67A#lecture05%5Frefslides) -  Although the entire class was spent working on the Step 2 script, I briefly referenced a few slides, which I've included here.

[Homework: Introduction to the Tidyverse (~4hrs)](https://www.datacamp.com/courses/introduction-to-the-tidyverse) - due before the start of class on March 6th.


## Overview

We'll begin this class by filtering and normalzing our data, all while using the ggplot2 graphing package to visualize the impact these changes have our data. You'll also be introduced to Hadley Wickham's philosophy of 'tidy data' by using the [dplyr](http://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html) package along with other tools from his [tidyverse](https://www.tidyverse.org/) of R packages to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data sets quickly and easily using simple commands in R.

## Goals

* Start/finish step 2 script
* Begin to explore our data in R using ggplot2
* Discuss basics of Tidy data and the reshape2 package
* Talk briefly about color palettes in R
* Filter data to remove lowly expressed genes
* Normalize data

## Code

[Step 2 script](http://DIYtranscriptomics.github.io/Code/files/Step2_dataWrangling.R)

## Software

[Sip](https://sipapp.io/) - I discussed this simple but excellent color picking software in class. The program costs about $10, but provides a convenient and powerful way to assemble color palettes for use in R.

## Reading

[Tidy Data](http://vita.had.co.nz/papers/tidy-data.pdf) - Hadley Wickham (author of Tidyverse packages and Chief Scientist at RStudio) describes the philosophy of tidy data in this paper. 

[Grammar of graphics](https://vita.had.co.nz/papers/ggplot2-wires.pdf) - Another paper by Hadley Wickham.  This one explains the rationale behind ggplot2.

[original TMM normalization manuscript](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25).

[catalog of R graphs](http://shiny.stat.ubc.ca/r-graph-catalog/) - Take a look at some of the various ways to graph your data and the underlying R code in this

[R Graphics Cookbook](http://hdl.library.upenn.edu/1017.12/1675994) - If you end up using R to make a lot of graphs, you will find the to be an important reference. It's available free to UPenn folks as an Ebook.

Colors palettes are an often underappreciated aspect of making beautiful and informative plots in R. You can access a suite of color palettes using the [RColorBrewer package](http://colorbrewer2.org).  These palettes can be viewed in [this cheatsheet](http://DIYtranscriptomics.github.io/Reading/files/colorbrewerPalettes.pdf).  Unfortunately, these standard palettes often don't cut it, and you'll need custom palettes. For this, I love using [Sip](https://sipapp.io/) to pick, organize and access color palettes. 