---
title: 'Functional enrichment analysis'
subtitle: 'Lecture 12 • 2019.11.13 • Levin 111 (SAIL classroom) • 3-5pm'
date: 2019-01-01 00:00:00
description: Now that you've identified differentially expressed genes, what do they mean and how do you begin to elucidate the biological pathways governed by these genes?  To address this question, in this class you'll learn how to carry out functional enichment analyses using Gene Ontology and Gene Set Enrichment methods.  You'll also explore different options for how to present your functional enrichment results.
featured_image: '/images/connections.png'
---

<div style="text-align: right"> Image credit: 'Connections' by Marsha Glickman </div>

[Lecture slides](https://www.icloud.com/keynote/0Naa63iYXqX4nTRUq2PF0Rfpg#Lecture11%5FFunctionalEnrichment)


## Overview

Now that you've identified differentially expressed genes, what do they mean and how do you begin to elucidate the biological pathways governed by these genes?  Toward this end, you will learn how to carry out functional enichment analyses using [Gene Ontology](http://geneontology.org/) and [Gene Set Enrichment Analysis (GSEA)](http://software.broadinstitute.org/gsea/index.jsp).  You'll also explore  different options for how to present your functional enrichment results graphically.

## Goals

* carry out Gene Ontology (GO) enrichment analysis
* carry out a GSEA enrichment analysis
* understand the difference between GO and GSEA
* understand the MSigDB resource

## Code

[Step 7 script](http://DIYtranscriptomics.github.io/Code/files/Step7_functionalEnrichment.R)

## Reading

[The what, where, how and why of gene ontology -- a primer for bioinformaticians.](http://DIYtranscriptomics.github.io/Reading/files/GO.pdf)  Briefings in Bioinformatics, Feb 2011

[A nice blog post on the hypergeometric test and Fisher's exact test](httP;//mengnote.blogspot.com/2012/12/calculate-correct-hypergeometric-p.html) - these statistical tests are at the core of many functional enrichment approaches.

[original 2003 Nat. Methods paper describing Gene Set Enrichment Analysis (GSEA)](http://DIYtranscriptomics.github.io/Reading/files/Mootha2003_GSEA.pdf), and the [2005 PNAS paper](http://mootha.med.harvard.edu/PubPDFs/Subramanian2005.pdf) that formally detailed its usage.

Not a fan of using the Broad Inst. GSEA program?  You can carry out self-contained and competitive GSEA in R using [ROAST](http://DIYtranscriptomics.github.io/Reading/files/ROAST.pdf) and [CAMERA](http://DIYtranscriptomics.github.io/Reading/files/CAMERA.pdf), respectively.

[Gene Set VARIATION Analysis (GSVA)](http://DIYtranscriptomics.github.io/Reading/files/GSVA.pdf) -  I find GSVA useful for producing GSEA-type results across a heterogeneous dataset (like a large cohort of patients).

[The Molecular Signatures Database (MSigDB)](http://software.broadinstitute.org/gsea/msigdb)

[2016 Immunity Paper describing the creation of Immunological Signatures' collection (C7)](http://DIYtranscriptomics.github.io/Reading/files/ImmuneSigDB.pdf).

[mouse and human specific gene signature databases](http://bioinf.wehi.edu.au/software/MSigDB/)

[GSEA file formats page](http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats).

You know how I feel about Venn diagrams, so if you're interested in exploring interactions between many groups of genes, have a look at [this Nature Methods paper](http://DIYtranscriptomics.github.io/Reading/files/upSet_plot.pdf), the accompanying R package, [UpSetR](https://cran.r-project.org/web/packages/UpSetR/README.html), as well as the [UpSet website](http://caleydo.org/tools/upset/).  Note, there's a shiny app for this as well!

Here are my Datagraph templates for making [enrichment bubble plots](https://drive.google.com/drive/folders/1J1Fvw-73BjXYnStvAv7XIkK3FqI6dsMe?usp=sharing) and [GSEA enrichment plots](https://drive.google.com/drive/folders/1uzP1EvJ1c5qCMZI_SwVa7Lz8qUa1OubM?usp=sharing)


## Lecture video

<iframe src="https://player.vimeo.com/video/374063512" width="640" height="564" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>