<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your lab or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title>Making your analysis portable and reproducible – DIY.transcriptomics</title>
	<meta name="description" content="Reproducing an analysis requires more than just code. You need the original raw data, access to the appropriate programming languages, and application specific packages (and often specific versions of these packages).  This poses a major impediment to reproducibility, even for researchers with a background in bioinformatics.  To address this challenge, you'll learn how to 'containerize'  your data, scripts and software, making it easy to share and rerun an entire analysis with the push of a button.">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="Making your analysis portable and reproducible – DIY.transcriptomics">
	<meta name="twitter:description" content="Reproducing an analysis requires more than just code. You need the original raw data, access to the appropriate programming languages, and application specific packages (and often specific versions of these packages).  This poses a major impediment to reproducibility, even for researchers with a background in bioinformatics.  To address this challenge, you'll learn how to 'containerize'  your data, scripts and software, making it easy to share and rerun an entire analysis with the push of a button.">
	<meta name="twitter:image:src" content="http://localhost:4000/images/capsule.jpg">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="Making your analysis portable and reproducible – DIY.transcriptomics" />
	<meta property="og:description" content="Reproducing an analysis requires more than just code. You need the original raw data, access to the appropriate programming languages, and application specific packages (and often specific versions of these packages).  This poses a major impediment to reproducibility, even for researchers with a background in bioinformatics.  To address this challenge, you'll learn how to 'containerize'  your data, scripts and software, making it easy to share and rerun an entire analysis with the push of a button." />
	<meta property="og:image" content="http://localhost:4000/images/capsule.jpg" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/project/class-15">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Modules</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/lab/" class="menu__list__item__link">Labs</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/videos" class="menu__list__item__link">Videos</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="Making your analysis portable and reproducible – DIY.transcriptomics">

			<section class="hero hero--single">

	<div class="hero__image" style="background-image: url(/images/capsule.jpg)">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1>Making your analysis portable and reproducible</h1>
		<p>Module 13 • 2hrs</p>

	</div>

</section>

<section class="single">

	<div class="wrap">

		<article class="single-post">

			<div style="text-align: right"> Image credit: 'Emoh: sculpture / time capsule &amp; temporary home' by Loren Naji </div>

<p><a href="https://www.icloud.com/keynote/0gPGSkggD-NUmJUg9tSyFis2w#Lecture13%5Freproducibility">Lecture slides on iCloud</a></p>

<h2 id="overview">Overview</h2>

<p>Reproducing an analysis requires more than just code. You need the original raw data, access to the appropriate programming languages, and application specific packages (and often specific versions of these packages).  This poses a major impediment to reproducibility, even for researchers with a background in bioinformatics.  To address this challenge, you’ll learn how to ‘containerize’  your data, scripts and software, making it easy to share and rerun an entire analysis with the push of a button.</p>

<h2 id="learning-objectives">Learning objectives</h2>

<ul>
  <li>Learn how to make your research analyses reproducible</li>
  <li>Create a reproducible package environment with <a href="https://rstudio.github.io/renv/articles/renv.html">renv</a></li>
  <li>Share your project via <a href="https://github.com/">GitHub</a> and git</li>
  <li>Understand how to streamline code using custom R functions.</li>
  <li>Share your work as an R package</li>
  <li>Discuss the basics of Docker and containerized software</li>
  <li>Use Code Ocean to access the entire DIY course as a reproducible computing environment</li>
</ul>

<h2 id="what-you-need-to-do">What you need to do</h2>

<ul>
  <li>Sign-up for an account on <a href="https://codeocean.com/">Code Ocean</a> (<em>be sure</em> to use the same email address from your DataCamp login).  You’ll get 15 compute hrs/month for free.</li>
  <li>Sign-up for a free <a href="https://github.com/">GitHub</a> account (doesn’t matter which email you use)</li>
  <li><a href="https://drive.google.com/open?id=13HhGnxAjCJilQNFHrIkBjGceXP6hZmKQ">Download this gitignore file</a> - useful for updating your own .gitignore file in a project repo</li>
  <li><a href="http://DIYtranscriptomics.github.io/Code/files/buildPkg.R">Download this script</a> that walks through how to turn any analysis project into an R package. You may also want this <a href="http://DIYtranscriptomics.github.io/Code/files/pkgDocs.txt">text file</a> as a simple starting point for data documentation, and <a href="http://DIYtranscriptomics.github.io/Code/files/DIYprofile.R">this function file</a> also as an example.</li>
</ul>

<h2 id="lecture-video">Lecture video</h2>

<h3 id="part-1---reproducibility-via-the-renv-package">Part 1 - Reproducibility via the renv package</h3>

<iframe src="https://player.vimeo.com/video/426273940" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<h3 id="part-2---connecting-your-project-to-github">Part 2 - Connecting your project to GitHub</h3>

<iframe src="https://player.vimeo.com/video/426287346" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<h3 id="part-3---code-ocean-capsules-for-full-reproducibility-in-publications">Part 3 - Code Ocean capsules for full reproducibility in publications</h3>

<iframe src="https://player.vimeo.com/video/426535680" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<h3 id="part-4---keeping-your-code-clean-via-custom-functions">Part 4 - Keeping your code clean via custom functions</h3>

<iframe src="https://player.vimeo.com/video/426552841" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<h3 id="part-5---how-to-turn-your-analysis-project-into-a-stand-alone-r-package">Part 5 - How to turn your analysis project into a stand alone R package</h3>

<iframe src="https://player.vimeo.com/video/427243128" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<hr />

<h2 id="code-ocean">Code Ocean</h2>

<p>We’ll use Code Ocean to interact with a dockerized container that packs all the code, data and software from the course into one reproducible and web-accessible environment.  Simply login (or set-up a free account if you don’t already have one) and you’ll be able to re-run the entire course in a matter of minutes, without any software installation or data download.  Your first run may take ~15min, since the full computing environment must be being built, but subsequent runs will be much faster.  Note that this capsule includes raw fastq files, kallisto outputs, all of MSigDB for running GSEA, and the entire ARCHS4 database for interrogating ~700,000 publically available mouse and human RNAseq datasets.  Have fun adapting this capsule for your own analyses!</p>

<script src="https://codeocean.com/widget.js?slug=1031026" async=""></script>

<h2 id="reading">Reading</h2>

<p><a href="https://happygitwithr.com/">Happy Git and GitHub with RStudio</a> - Jenny Bryan and team walk through every step of how to install git, connect to GitHub and access version control from within RStudio.</p>

<p>There’s a lot of reading material for how to get started making functions and packages.  Beyond the extensive and very well written <a href="http://r-pkgs.had.co.nz/">book on building R packages</a> and excellent documentation for the <a href="https://usethis.r-lib.org/">usethis package</a>, you may also want to check out some great lab posts on making R packages (<a href="https://masalmon.eu/2017/12/11/goodrpackages/">here</a>, <a href="https://r-mageddon.netlify.app/post/writing-an-r-package-from-scratch/">here</a>, <a href="https://tinyheero.github.io/jekyll/update/2015/07/26/making-your-first-R-package.html">here</a>, and <a href="https://kbroman.org/pkg_primer/">here</a>).</p>

<p><a href="http://DIYtranscriptomics.github.io/Reading/files/codeOcean_whitepaper.pdf">Code Ocean whitepaper</a> - describes the need for better tools for reproducible research and introduces their cloud-based computational platform for addressing this need.</p>

<p><a href="https://docker-curriculum.com/">Intro to Docker</a> - Code Ocean is based on Docker, a free and open-source tool that allows you to ‘build, share and run applications anywhere’.</p>

<p><a href="https://docs.google.com/document/d/1baA8IRFDZE1rJNB3SYk7Aksvuz5AJ9W4HG2iqYoPt-8/edit?usp=sharing">Code Ocean On-boarding document</a> - Step-by-step details for how to set-up your own capsule.</p>

<p><a href="https://stm.sciencemag.org/content/11/519/eaax4204">Our recent paper</a>, showing a code capsule embedded directly in the joural webpage (a first for any AAAS journal).</p>

<p><a href="https://colab.research.google.com/">Google Collaboratory</a> - Write, edit and share Python code directly in your browser</p>

<hr />



		</article>

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">Teaching students how to use open-source tools to analyze RNAseq data since 2015.</p>

		<div class="footer__copyright">
			<span>© 2021 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	

</body>

</html>