<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your lab or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title>Data exploration – DIY.transcriptomics</title>
	<meta name="description" content="In this class you'll learn about a variety of approaches exploring your data.  You'll use multivariate statistical approaches such as Principal Component Analysis (PCA) to understand sources of variance in our data, while continuing to build your plotting skills by using ggplot2 to graph the results of PCA analyses. You'll also learn how to use the dplyr package to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data sets quickly and easily using simple commands in R.  We’ll discuss common missteps and how to identify sources of bias in transcriptional data sets.">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="Data exploration – DIY.transcriptomics">
	<meta name="twitter:description" content="In this class you'll learn about a variety of approaches exploring your data.  You'll use multivariate statistical approaches such as Principal Component Analysis (PCA) to understand sources of variance in our data, while continuing to build your plotting skills by using ggplot2 to graph the results of PCA analyses. You'll also learn how to use the dplyr package to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data sets quickly and easily using simple commands in R.  We’ll discuss common missteps and how to identify sources of bias in transcriptional data sets.">
	<meta name="twitter:image:src" content="http://localhost:4000/images/explorer.jpg">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="Data exploration – DIY.transcriptomics" />
	<meta property="og:description" content="In this class you'll learn about a variety of approaches exploring your data.  You'll use multivariate statistical approaches such as Principal Component Analysis (PCA) to understand sources of variance in our data, while continuing to build your plotting skills by using ggplot2 to graph the results of PCA analyses. You'll also learn how to use the dplyr package to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data sets quickly and easily using simple commands in R.  We’ll discuss common missteps and how to identify sources of bias in transcriptional data sets." />
	<meta property="og:image" content="http://localhost:4000/images/explorer.jpg" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/project/class-07">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Modules</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/lab/" class="menu__list__item__link">Labs</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/videos" class="menu__list__item__link">Videos</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="Data exploration – DIY.transcriptomics">

			<section class="hero hero--single">

	<div class="hero__image" style="background-image: url(/images/explorer.jpg)">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1>Data exploration</h1>
		<p>Module 7 • 2hrs</p>

	</div>

</section>

<section class="single">

	<div class="wrap">

		<article class="single-post">

			<div style="text-align: right"> Image credit: 'Maps' by Cara Barer</div>

<p><a href="https://www.icloud.com/keynote/0yQPc4lJXR9owaIo8fNO5IQxg#Lecture7%5FdataExploration">Lecture slides on iCloud</a></p>

<p><a href="https://www.datacamp.com/courses/data-visualization-with-ggplot2-1">Homework #3: Data visualization with ggplot2 - part 1 (~4hrs)</a> - due by start of class next Wednesday, May 13th.</p>

<h2 id="overview">Overview</h2>

<p>In this class you’ll learn about a variety of approaches exploring your data.  You’ll use multivariate statistical approaches such as Principal Component Analysis (PCA) to understand sources of variance in our data, while continuing to build your plotting skills by using ggplot2 to graph the results of PCA analyses. You’ll also learn how to use the dplyr package to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data sets quickly and easily using simple commands in R.  We’ll discuss common missteps and how to identify sources of bias in transcriptional data sets.</p>

<h2 id="learning-objectives">Learning objectives</h2>

<ul>
  <li>Start and finish Step 3 script</li>
  <li>Discuss basics of multivariate statistical analysis</li>
  <li>Carry out hierarchical clustering of samples</li>
  <li>Discuss and perform principal component analyses (PCA)</li>
  <li>Produce ‘small multiples’ plot</li>
  <li>Use standard dplyr ‘verbs’ to quickly query our data</li>
  <li>Produce interactive graphics using the plotly package</li>
  <li>Produce interactive tables with DT package</li>
</ul>

<h2 id="code">Code</h2>

<p><a href="http://DIYtranscriptomics.github.io/Code/files/Step3_multivariate.R">Step 3 script</a></p>

<hr />

<h2 id="lecture-video">Lecture video</h2>

<h3 id="part-1---discussion-of-multivariate-data-dimensional-reduction-via-pca-and-starting-our-step-3-script">Part 1 - Discussion of multivariate data, dimensional reduction via PCA, and starting our Step 3 script</h3>

<iframe src="https://player.vimeo.com/video/415584939" width="640" height="267" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<h3 id="part-2---plotting-pca-results-and-small-multiples">Part 2 - Plotting PCA results and small multiples</h3>

<iframe src="https://player.vimeo.com/video/415642798" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<h3 id="part-3---producing-interactive-tables-and-plots">Part 3 - Producing interactive tables and plots</h3>

<iframe src="https://player.vimeo.com/video/415644355" width="640" height="268" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>

<hr />

<h2 id="reading">Reading</h2>

<p><a href="https://doi.org/10.1371/journal.pcbi.1006907">Ten quick tips for effective dimensionality reduction</a> - a absolute must-read for understanding data exploration methods.</p>

<p><a href="http://distill.pub/2016/misread-tsne/">lab post describing T-SNE</a> - I mentioned various unsupervised linear methods for dimensional reduction of your data (PCA, MDS).  T-SNE and UMAP are <em>non-linear</em> unsupervised methods that have become popular for representing single-cell RNAseq data and flow cytometry data.</p>

<p><a href="http://DIYtranscriptomics.github.io/Reading/files/TSNE.pdf">Original T-SNE paper</a>.</p>

<p><a href="https://www.nature.com/articles/nbt.4314">UMAP paper</a> - A new algorithm, called uniform manifold approximation and projection (UMAP) has been recently published and is gaining popularity in single cell RNAseq and flow cytometry analysis.  UMAP is proposed to preserve as much of the local and more of the global data structure than t-SNE, with a shorter run time.</p>

<h2 id="other-videos">Other videos</h2>

<h3 id="josh-starmer-does-a-fantastic-job-breakding-down-the-math-behind-pca-in-his-statquest-series">Josh Starmer does a fantastic job breakding down the math behind PCA in his StatQuest series</h3>

<iframe width="560" height="315" src="https://www.youtube.com/embed/FgakZw6K1QQ" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>


		</article>

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">Teaching students how to use open-source tools to analyze RNAseq data since 2015.</p>

		<div class="footer__copyright">
			<span>© 2021 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	

</body>

</html>