<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your blog or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title>Functional enrichment analysis – DIY.transcriptomics</title>
	<meta name="description" content="Now that you've identified differentially expressed genes, what do they mean and how do you begin to elucidate the biological pathways governed by these genes?  Toward this end, you will learn how to carry out functional enichment analyses using [Gene Ontology].  You'll also see some different options for how to present your functional enrichment results.">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="Functional enrichment analysis – DIY.transcriptomics">
	<meta name="twitter:description" content="Now that you've identified differentially expressed genes, what do they mean and how do you begin to elucidate the biological pathways governed by these genes?  Toward this end, you will learn how to carry out functional enichment analyses using [Gene Ontology].  You'll also see some different options for how to present your functional enrichment results.">
	<meta name="twitter:image:src" content="http://localhost:4000/images/connections.png">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="Functional enrichment analysis – DIY.transcriptomics" />
	<meta property="og:description" content="Now that you've identified differentially expressed genes, what do they mean and how do you begin to elucidate the biological pathways governed by these genes?  Toward this end, you will learn how to carry out functional enichment analyses using [Gene Ontology].  You'll also see some different options for how to present your functional enrichment results." />
	<meta property="og:image" content="http://localhost:4000/images/connections.png" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/project/lecture-12">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Syllabus</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="Functional enrichment analysis – DIY.transcriptomics">

			<section class="hero hero--single">

	<div class="hero__image" style="background-image: url(/images/connections.png)">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1>Functional enrichment analysis</h1>
		<p>Lecture 12 • 2019.04.10 • Hill 130</p>

	</div>

</section>

<section class="single">

	<div class="wrap">

		<article class="single-post">

			<h2 id="overview">Overview</h2>

<p>Now that you’ve identified differentially expressed genes, what do they mean and how do you begin to elucidate the biological pathways governed by these genes?  Toward this end, you will learn how to carry out functional enichment analyses using <a href="http://geneontology.org/">Gene Ontology</a>.  You’ll also see some different options for how to present your functional enrichment results.</p>

<h2 id="goals">Goals</h2>

<ul>
  <li>carry out Gene Ontology (GO) enrichment analysis</li>
  <li>carry out a GSEA enrichment analysis</li>
  <li>understand the difference between GO and GSEA</li>
  <li>understand the MSigDB resource</li>
</ul>

<h2 id="code">Code</h2>

<p><a href="http://DIYtranscriptomics.github.io/Code/files/Step7_geneSetAnalysis.R">Step 7 script</a></p>

<h2 id="reading">Reading</h2>

<p><a href="http://DIYtranscriptomics.github.io/Reading/files/GO.pdf">The what, where, how and why of gene ontology – a primer for bioinformaticians.</a>  Briefings in Bioinformatics, Feb 2011</p>

<p><a href="httP;//mengnote.blogspot.com/2012/12/calculate-correct-hypergeometric-p.html">A nice blog post on the hypergeometric test and Fisher’s exact test</a> - these statistical tests are at the core of many functional enrichment approaches.</p>

<p><a href="http://DIYtranscriptomics.github.io/Reading/files/Mootha2003_GSEA.pdf">original 2003 Nat. Methods paper describing Gene Set Enrichment Analysis (GSEA)</a>, and the <a href="http://mootha.med.harvard.edu/PubPDFs/Subramanian2005.pdf">2005 PNAS paper</a> that formally detailed its usage.</p>

<p>Not a fan of using the Broad Inst. GSEA program?  You can carry out self-contained and competitive GSEA in R using <a href="http://DIYtranscriptomics.github.io/Reading/files/ROAST.pdf">ROAST</a> and <a href="http://DIYtranscriptomics.github.io/Reading/files/CAMERA.pdf">CAMERA</a>, respectively.</p>

<p><a href="http://DIYtranscriptomics.github.io/Reading/files/GSVA.pdf">Gene Set VARIATION Analysis (GSVA)</a> -  I find GSVA useful for producing GSEA-type results across a heterogeneous dataset (like a large cohort of patients).</p>

<p><a href="http://software.broadinstitute.org/gsea/msigdb">The Molecular Signatures Database (MSigDB)</a></p>

<p><a href="http://DIYtranscriptomics.github.io/Reading/files/ImmuneSigDB.pdf">2016 Immunity Paper describing the creation of Immunological Signatures’ collection (C7)</a>.</p>

<p><a href="http://bioinf.wehi.edu.au/software/MSigDB/">mouse and human specific gene signature databases</a></p>

<p><a href="http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats">GSEA file formats page</a>.</p>

<p>You know how I feel about Venn diagrams, so if you’re interested in exploring interactions between many groups of genes, have a look at <a href="http://DIYtranscriptomics.github.io/Reading/files/upSet_plot.pdf">this Nature Methods paper</a>, the accompanying R package, <a href="https://cran.r-project.org/web/packages/UpSetR/README.html">UpSetR</a>, as well as the <a href="http://caleydo.org/tools/upset/">UpSet website</a>.  Note, there’s a shiny app for this as well!</p>

<p>Here are my Datagraph templates for making <a href="https://drive.google.com/drive/folders/1J1Fvw-73BjXYnStvAv7XIkK3FqI6dsMe?usp=sharing">enrichment bubble plots</a> and <a href="https://drive.google.com/drive/folders/1uzP1EvJ1c5qCMZI_SwVa7Lz8qUa1OubM?usp=sharing">GSEA enrichment plots</a></p>


		</article>

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">A course on using open-source tools to analyze RNAseq data.</p>

		<div class="footer__copyright">
			<span>© 2019 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	


</body>

</html>