<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your lab or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title>Data – DIY.transcriptomics</title>
	<meta name="description" content="Access to fastq files and info for the dataset used for the course.">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="Data – DIY.transcriptomics">
	<meta name="twitter:description" content="Access to fastq files and info for the dataset used for the course.">
	<meta name="twitter:image:src" content="http://localhost:4000/images/corte.jpg">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="Data – DIY.transcriptomics" />
	<meta property="og:description" content="Access to fastq files and info for the dataset used for the course." />
	<meta property="og:image" content="http://localhost:4000/images/corte.jpg" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/data">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Modules</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/lab/" class="menu__list__item__link">Labs</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/videos" class="menu__list__item__link">Videos</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="Data – DIY.transcriptomics">

			<section class="hero hero--single">

	<div class="hero__image" style="background-image: url(/images/corte.jpg)">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1>Data</h1>
		<p>Access to fastq files and info for the dataset used for the course.</p>

	</div>

</section>

<section class="single">

	<div class="wrap">

		<article class="single-post">

			<h2 id="course-dataset">Course dataset</h2>

<p>The course dataset comes from a collaboration between my lab, Phil Scott’s lab, and our colleagues in Salvador, Brazil, and was <a href="https://doi.org/10.1126/scitranslmed.aax4204">recently published</a> by none other than the course TA, Camila!  This is an RNAseq dataset from skin biopsies obtained from patients with cutaneous leishmaniasis, a parasitic disease endemic in Brazil and other areas of South America.  You’ll be working with data from 5 patients with this disease and 5 healthy endemic controls.</p>

<p>This course offers multiple ‘on-ramps’, or entry points, for starting an analysis.  Below are a few of these options.</p>

<hr />

<h3 id="option-1-preferred--download-raw-data-and-accessory-files">Option 1 (preferred)- Download raw data and accessory files</h3>

<p><a href="https://www.dropbox.com/sh/df58trgab010s55/AAAQ86KkKPzuqvGG-YoeISNEa?dl=0">fastq files</a> – You will need about <strong>30Gb</strong> of storage space on your harddrive to accomodate this download.  <em>please do not uncompress these files (leave them as .gz files)</em>.</p>

<p><a href="https://www.dropbox.com/s/c1vy2bdg4fynk7e/studydesign.txt?dl=0">study design file</a> - A simple text file that contains one row for each sample and columns for each variable in the study.</p>

<p><a href="https://www.dropbox.com/s/h0rqothvvbu3hzm/readMapping.sh?dl=0">shell script</a> – To carry out read mapping and QC analysis for multiple samples</p>

<hr />

<h3 id="option-2--download-data-already-mapped">Option 2 – Download data already mapped</h3>

<p>In the event that you don’t have enough free harddrive space to download the raw fastq files in Option 1 above, or if you have any problems installing or using Kallisto to map this raw data, you can bypass the Kallisto step and download mapped data to start working immediately.</p>

<p><a href="https://www.dropbox.com/s/94klzelk2w2w2mj/mappedReads.zip?dl=0">Kallisto output</a> -  These files are available as a single compressed zip file.  Download and unzip this file to reveal 10 folders (each containing the Kallisto output from mapping each of the 10 fastq files above), along with 10 log files – one for each sample that was mapped.  You may notice that each folder contains several files.  Please do not move or rename these files.  During the course, we will discuss these Kallisto outputs.</p>

<hr />

<h2 id="can-i-work-with-my-own-data-during-the-course">Can I work with my own data during the course?</h2>

<p>Yes!  We encourage students to work with their own data during the course.  Nothing will keep you more enagaged and crystalize the content of the course quite like working on questions you <em>actually</em> care about.  That said, it is not a requirement that you come with data in hand.  Throughout the course, I will demonstrate every step in the analysis using the dataset above. If you are working with your own dataset and run into problems, I strongly recommend that you switch to the course dataset so you don’t fall behind.  You can always return to working on your own data once you have a functional pipeline and understand the process.</p>

<hr />

<h2 id="where-can-i-access-public-rna-seq-data">Where can I access public RNA-seq data</h2>

<p>In the second half of the course, you’ll learn how to access hundreds of thousands of samples already pre-mapped using Kallisto!  These are available through several different resources, which we’ll discuss, but we’ll focus on the <a href="https://amp.pharm.mssm.edu/archs4/index.html">ARCHS4 database from Avi Ma’ayan’s lab</a>.  You’ll want to download the compressed ARCHS4 databases for mouse samples <a href="https://www.dropbox.com/s/crz1ctpqhbpb1ox/mouse_matrix_v8.h5?dl=0">here</a>.  Note that this download is ~7GB of data.</p>


		</article>

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">Teaching students how to use open-source tools to analyze RNAseq data since 2015.</p>

		<div class="footer__copyright">
			<span>© 2021 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	

</body>

</html>