<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your lab or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title>Help – DIY.transcriptomics</title>
	<meta name="description" content="A collection of local (UPenn), regional (Philly), and online places where you can go to learn more about using R/bioconductor.">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="Help – DIY.transcriptomics">
	<meta name="twitter:description" content="A collection of local (UPenn), regional (Philly), and online places where you can go to learn more about using R/bioconductor.">
	<meta name="twitter:image:src" content="http://localhost:4000">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="Help – DIY.transcriptomics" />
	<meta property="og:description" content="A collection of local (UPenn), regional (Philly), and online places where you can go to learn more about using R/bioconductor." />
	<meta property="og:image" content="http://localhost:4000" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/help">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Syllabus</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="Help – DIY.transcriptomics">

			<section class="hero hero--single">

	<div class="hero__image" style="background-image: url()">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1>Help</h1>
		<p>A collection of local (UPenn), regional (Philly), and online places where you can go to learn more about using R/bioconductor.</p>

	</div>

</section>

<section class="single">

	<div class="wrap">

		<article class="single-post">

			<h2 id="getting-help-from-the-tas">Getting help from the TAs</h2>

<p>Camila Amorim and Alex Berry are Teaching Assistants for the course.  Please don’t hesitate to reach out to them via the course <a href="https://join.slack.com/t/diytranscriptomics/shared_invite/enQtNDE1MTI2MzY2MDY5LWMxOWI2YWE5YmVjM2ZkMjM1ZjYxNTdkOWRkM2ZlNGI0OTYzMWQ2NDRmMjRjMTBhNTVkZjdjZTBkODI0ODFlNmY">Slack page</a> if you have questions.</p>

<p>1hr open TA help sessions are held every week in room H234 (in PennVet library) immediately before class at 2-3pm.</p>

<hr />

<h2 id="ebooks-on-general-rbioconductor">Ebooks on general R/bioconductor</h2>

<p><a href="http://r4ds.had.co.nz/">R for Data Science</a>, by Hadley Wickham, is really a wonderful (and free!) online book.  If you decide to read this, which I highly recommend, then be sure to complement with the <a href="https://jrnold.github.io/r4ds-exercise-solutions/">R4DS companion solutions manual</a>.</p>

<p><a href="http://hdl.library.upenn.edu/1017.12/1675994">R Graphics Cookbook</a> - an essential reference book for plotting data in R.</p>

<p><a href="http://DIYtranscriptomics.github.io/Reading/files/Cheatsheets_2019.pdf">RStudio Cheatsheets</a> - great 1-2pg summaries of the essential R/RStudio functions.  Print these out and keep them handy when you’re working in RStudio.</p>

<p><a href="http://web.stanford.edu/class/bios221/book/">Modern Statistics for Modern Biology</a> - A new book that just came out written two of the leaders in biostats, and it looks to be an excellent R-based guide to the topic, with many great coding examples.</p>

<hr />

<h2 id="finding-help-online">Finding help online</h2>

<p><a href="http://www.bioconductor.org/help/">Bioconductor’s help site</a> - full of great tutorials, links to videos, and  message board for more more complicated questions.</p>

<p><a href="https://www.biostars.org/">Biostars</a> - More general bioinformatics questions not pertaining to a specific R package are best addressed here.</p>

<p><a href="http://stackoverflow.com/">StackOverflow</a> - great place for programming questions that are not related specifically to bioinformatics.</p>

<hr />

<h2 id="getting-help-without-ever-leaving-rstudio">Getting help without ever leaving RStudio</h2>

<p><strong>Help within R</strong> - R has a comprehensive help system built right in! Type <code class="highlighter-rouge">?FunctionName</code> right in the R terminal (no quotes and replace FunctionName with the name of the function you need help with).</p>

<p><strong>package vignettes</strong> - Each Bioconductor package contains at least one vignette, a document that provides a task-oriented description of package functionality. Vignettes contain executable examples and are intended to be used interactively.  To access the vignette for a package, simply type <code class="highlighter-rouge">browseVignettes(packageName)</code> (without quotes)</p>

<p><strong>example datasets</strong> - type <code class="highlighter-rouge">data()</code> to see all available datasets for the packages you have loaded.  These can be extremely useful when learning to use new packages.</p>

<hr />

<h2 id="take-advantage-of-your-local-upenn-and-philly-community">Take advantage of your local UPenn and Philly community</h2>

<p>Below are a few examples of some of the great local resources.</p>

<p><a href="https://pennusergroup.wordpress.com">Use R! group at UPenn</a> - A groups of Penn staff, students and researchers who get together to improve their R knowledge and skills.  The group, which is open to R users of all levels, meets weekly on Wednesdays, 11am-noon, in the Weigle Informations Commons (WIC) seminar room of the Van Pelt-Dietrich Library Center.</p>

<p><a href="">CHOPR</a></p>

<p><a href="https://groups.google.com/forum/?hl=en#!forum/bushmancode">Weekly ‘Code Review’</a> - Led by Kyle Bittinger, a Computational Biologist and faculty at CHOP.  Each week one person presents some programming code they are working on for their research.  Attendees get a chance to hear how code was put together to achieve a goal, while the presenter gets feedback, collective troubleshooting, etc.  Less experienced programmers will have a chance to learn about design options when programming.  Group meets one Wednesdays @ 9am in Johnson 207.  Contact Kyle by email at <a href="mailto:kylebittinger@gmail.com">kylebittinger@gmail.com</a> to be added to the mailing list.</p>

<p><a href="http://pathbio.med.upenn.edu/gs/cytomics/">Cytomic Data Analysis Workshop</a> - Led by Wade Rogers, the Director of Computational Biology and Research Informatics for the Path BioResource at the Perelman School of Medicine.  Given the pervasive nature of flow cytometry in all aspects of biomedical research, the mutiparameter nature of flow experiments, and the increasing abundance of clinical metadata, it only makes sense to leverage the power of R for analyzing the rich datasets produced by flow cytometry.  Wade is an expert user of R/bioconductor for flow analysis, and has been very active in tool development in this area, including the development of a <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/flowFP/inst/doc/flowFP_HowTo.pdf">‘fingerprinting’</a> method for analysing flow data.  Contact Wade by email at <a href="mailto:rogersw@mail.med.upenn.edu">rogersw@mail.med.upenn.edu</a> for more information on the workshop.</p>

<p><a href="http://rladiesphilly.org/">R-Ladies Philly</a> is part of a world-wide organization to promote gender diversity in the R community.  R-Ladies Philly is a free, community-driven group that is open to all levels of R users. While the group name does have ‘Ladies’ in the title, they welcome all individuals who share their values.  Monthly meetings are skills-focused, with encouragement of interdisciplinary statistical and coding methods.</p>

<p><a href="https://www.girldevelopit.com/chapters/philadelphia">Girl Develop It</a> - Philadelphia has a local chapter of GDI that hosts speakers, workshops and social events aimed at bringing together and empowering women to learn web and software development in a judgment-free environment.</p>

<p><a href="http://femmehacks.io/">FemmeHacks</a> is a all-women collegiate hackathon held in Philadelphia at UPenn’s Huntsman Hall, and is hosted by the Penn <a href="www.seas.upenn.edu/~wics">Women in Computer Science (WiCS)</a>.  This year’s hackathon will take place in February.  Friday, Feb 24th will consist of social mixers, workshops, and teams will be formed at Penn Engineering (220 South 33rd St), with the hackthon taking place the following day on Saturday the 25th at the Pennovation Center (3401 Grays Ferry Ave)</p>

<hr />

<h2 id="take-an-online-course-to-hone-and-expand-your-data-science-skills">Take an online course to hone and expand your data science skills</h2>

<p><a href="https://www.datacamp.com">Datacamp tutorials</a> - You have access to the full content of Datacamp <em>for free</em> during the course, so why not work toward a certificate in data science using R.</p>

<p><a href="https://www.coursera.org/course/genbioconductor">Bioconductor for Genomic Data Science</a> - This is the fifth course in the Genomic Data Science Specialization from Johns Hopkins University. (see below).  Learn to use tools from the Bioconductor project to perform analysis of genomic data.</p>

<p><a href="https://www.coursera.org/specializations/genomics?utm_medium=courseDescripTop">Coursera’s specialization in Genomic Data Science</a>. Takes you through 7 courses, each lasting 1 month, that cover topics relevant to the analysis of data generated by biomedical research approaches.</p>

<p><a href="http://bit.ly/1NLDqnn">Coursera’s specialization in Data Science</a> - Brought to you by Johns Hopkins and Coursera, this 10 course ‘degree’ spans nearly a year, and covers takes a much more broad approach to understanding data than the genomic specialization described above.  This specialization offers more on statistics, modeling, deploying data to the web, etc.  In a way, I feel that my course on transcriptomics is a blend of the Coursera Genomic Data Science and Data Science Specializations.</p>

<p><a href="https://www.coursera.org/specialization/systemsbiology/6?utm_medium=courseDescripTop">Coursera’s specialization in Systems Biology</a> - Yet another specialization offered through Coursera, this time in partnership with the Icahn School of Medicine at Mount Sinai and Coursera, this 5 course ‘degree’ covers the basics of network analysis and integrating diverse data types.</p>

<p><a href="rafalab.github.io/pages/harvardx.html">HarvardX’s course of Biomedical Data Science</a> - and excellent and free online course taught by the talented <a href="http://rafalab.dfci.harvard.edu/">Rafael Irizarry</a></p>


		</article>

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">Teaching students how to use open-source tools to analyze RNAseq data since 2015.</p>

		<div class="footer__copyright">
			<span>© 2020 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	

</body>

</html>