<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your blog or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title>DIY.transcriptomics – RNAseq course.</title>
	<meta name="description" content="A full course covering best practices for RNAseq data analysis, with a primary focus on empowering students to be independent in the use of lightweight and open-source software and the R/bioconductor environment.">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="DIY.transcriptomics – RNAseq course.">
	<meta name="twitter:description" content="A full course covering best practices for RNAseq data analysis, with a primary focus on empowering students to be independent in the use of lightweight and open-source software and the R/bioconductor environment.">
	<meta name="twitter:image:src" content="http://localhost:4000/images/network_white.jpg">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="DIY.transcriptomics – RNAseq course." />
	<meta property="og:description" content="A full course covering best practices for RNAseq data analysis, with a primary focus on empowering students to be independent in the use of lightweight and open-source software and the R/bioconductor environment." />
	<meta property="og:image" content="http://localhost:4000/images/network_white.jpg" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Syllabus</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="DIY.transcriptomics – RNAseq course.">

			<section class="hero">

	<div class="hero__image" style="background-image: url(/images/network_white.jpg)">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1>Welcome to DIY Transcriptomics</h1>
		<p>A full course covering best practices for RNAseq data analysis, with a primary focus on empowering students to be independent in the use of lightweight and open-source software and the R/bioconductor environment.  This course is currently offered as CAMB714 at the University of Pennsylvania's School of Veterinary Medicine.</p>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</section>

<section class="listing">

	<div class="wrap">

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/blankcanvas.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/before-course">Pre-course preparation</a></h2>
					<p class="post__subtitle">review prior to course start</p>
					<p class="post__description">Although there are no formal prerequisties for the course and no official course text book, there are a few important housekeeping things you'll need to take care of before the first class.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/illumina_flowcell.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-01">Intro to RNAseq technology and data</a></h2>
					<p class="post__subtitle">Lecture 1 • 2019.08.28 • Levin 111 (SAIL classroom)• 3-5pm</p>
					<p class="post__description">After a brief overview of the course, we'll spend the opening lecture talking about Illumina's 'Sequencing by Synthesis' technology, and walk through the steps involved in planning for a transcriptomics experiment.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/rstudio.png)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-02">Setting up your software environment</a></h2>
					<p class="post__subtitle">Lecture 2 • 2019.09.04 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">In the first half of this lecture we'll discuss the open-source, cross-platform R/bioconductor software and environment that will be used throughout the course, and each student will set-up their own laptop to be a powerful, stand-alone bioinformatics workstation.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/bear.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-03">Ultra-fast read mapping with Kallisto</a></h2>
					<p class="post__subtitle">Lecture 3 • 2019.09.11 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">In this class we'll finally get down to the business of using Kallisto for memory-efficient mapping of your raw reads.  You'll carry out this mapping in class, right on your laptop, while we discuss what's happening under the hood. During this process, we'll touch on a range of topics, from reference files, to command line basics, and using shell scripts for automation and reproducibility.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/abacus.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-04">Understanding RNAseq count data</a></h2>
					<p class="post__subtitle">Lecture 4 • 2019.09.18 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">Now that we've aligned our reads, it's time to discuss units for measuring gene expression.  We'll discuss differences between RPKM and TPM, and how these units relate to basic properties of your reference file and data.  We'll also discuss normalization within and between samples.  To conclude this class, we'll fire up RStudio and take a look at our first script.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/first_step.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-05">Starting your R workflow</a></h2>
					<p class="post__subtitle">Lecture 5 • 2019.09.25 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">We'll begin this class by reviewing how to access R packages and help documentation, as well as understanding the basic structure of an R script and RStudio project. We'll then access annotation data before reading our Kallisto results into R.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/tidyDNA.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-06">Wrangling gene expression data</a></h2>
					<p class="post__subtitle">Lecture 6 • 2019.10.02 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">We'll begin this class by filtering and normalzing our data, all while using the ggplot2 graphing package to visualize the impact these changes have our data. You'll also be introduced to Hadley Wickham's philosophy of 'tidy data' by using the dplyr package, expanding your understanding of tools within the Tidyverse.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/dataExplore.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-07">Data exploration</a></h2>
					<p class="post__subtitle">Lecture 7 • 2019.10.09 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">In this class you'll learn about a variety of approaches exploring your data.  You'll use multivariate statistical approaches such as Principal Component Analysis (PCA) to understand sources of variance in our data, while continuing to build your plotting skills by using ggplot2 to graph the results of PCA analyses. You'll also learn how to use the dplyr package to take control over our gene expression dataframes, allowing us to change, sort, filter, arrange and summarize large data sets quickly and easily using simple commands in R.  We’ll discuss common missteps and how to identify sources of bias in transcriptional data sets.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/DataDeluge.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-08">Accessing public data</a></h2>
					<p class="post__subtitle">Lecture 8 • 2019.10.16 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">In this class we’ll discuss how you can use R/Bioconductor to tap into vast amounts of RNAseq data available through the Sequence Read Archive (SRA) and Gene Expression Omnibus (GEO).</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/timepiece.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-09">Hackdash 1</a></h2>
					<p class="post__subtitle">Lecture 9 • 2019.10.23 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">You may have heard of a Hackathon before - an event that typically lasts several days and brings together people for a collaborative challenge in computer programming.  Well, we don't quite have time for a full Hackathon, so we'll do a mad dash to the finish line instead.  This 2hr in-class assignment will test your ability to use recent class material and code to solve a problem in a real RNAseq dataset.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/apples.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-10">Differential gene expression</a></h2>
					<p class="post__subtitle">Lecture 10 • 2019.10.30 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">The ultimate goal of most transcriptional profiling experiments is to identify differentailly expressed genes or transcripts.  In this class, we'll dig into differential expression using the popular and venerable Limma package in R. Finally, we'll continue to explore options for producing compelling plots from your differential expression results.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/bird_flock.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-11">Module identification</a></h2>
					<p class="post__subtitle">Lecture 11 • 2019.11.06 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">Coordinately expressed genes, or modules, often contain a more coherent functional signature.  We'll discuss strategies for clustering expression data to identify these modules, setting the stage for downstream function enrichment analysis to be covered in the next class.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/connections.png)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-12">Functional enrichment analysis</a></h2>
					<p class="post__subtitle">Lecture 12 • 2019.11.13 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">Now that you've identified differentially expressed genes, what do they mean and how do you begin to elucidate the biological pathways governed by these genes?  To address this question, in this class you'll learn how to carry out functional enichment analyses using Gene Ontology and Gene Set Enrichment methods.  You'll also explore different options for how to present your functional enrichment results.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/timepiece.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-13">Hackdash II</a></h2>
					<p class="post__subtitle">Lecture 13 • 2019.11.20 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">In this second Hackdash, you will have incorporate both differential gene expression analysis and downstream functional enrichment analysis to interroagte the human transcriptional response in the skin of patients infected with Leishmania braziliensis.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/breadcrumbs2.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-14">Documenting your analyses with Rmarkdown</a></h2>
					<p class="post__subtitle">Lecture 14 • 2019.12.04 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">In order to make your analysis pipeline transparent, in this class you'll use Rmarkdown and Knitr to wrap up all your code and outputs together in a dynamic document that can be placed in your lab notebook or published as a supplementary file in your manuscript.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/capsule.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-15">Making your analysis portable and reproducible with Docker</a></h2>
					<p class="post__subtitle">Lecture 15 • 2019.12.11 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">Reproducing an analysis requires more than just code. You need the original raw data, access to the appropriate programming languages, and application specific packages (and often specific versions of these packages).  This poses a major impediment to reproducibility, even for researchers with a background in bioinformatics.  To address this challenge, you'll learn how to 'containerize'  your data, scripts and software into a code timecapsule, making it easy to share and rerun an entire analysis with the push of a button.</p>
				</div>
			</div>

		</article>

		

		<article class="post">

			<div class="post__image-wrap">
				<div class="post__image" style="background-image: url(/images/pizza.jpg)"></div>
			</div>

			<div class="post__content-wrap">
				<div class="post__content">
					<h2 class="post__title"><a href="/project/lecture-16">Hackdash III</a></h2>
					<p class="post__subtitle">Lecture 16 • 2019.12.18 • Levin 111 (SAIL classroom) • 3-5pm</p>
					<p class="post__description">We will conclude the course with a third and final hackdash (and we'll eat pizza!).</p>
				</div>
			</div>

		</article>

		

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">Teaching students how to use open-source tools to analyze RNAseq data since 2015.</p>

		<div class="footer__copyright">
			<span>© 2019 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	


</body>

</html>