<!doctype html>

<html class="no-js" lang="en">

<head>


	<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

	DIY.transcriptomics

	Personal Theme by https://jekyllthemes.io
	Premium + free Jekyll themes for your blog or website.

	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->


	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<!-- Page Info -->
	<link rel="shortcut icon" href="/images/favicon.png">
	<title>Scripts – DIY.transcriptomics</title>
	<meta name="description" content="Access to all the scripts and code used throughout the course.">

	<!-- Twitter Card -->
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:title" content="Scripts – DIY.transcriptomics">
	<meta name="twitter:description" content="Access to all the scripts and code used throughout the course.">
	<meta name="twitter:image:src" content="http://localhost:4000/images/code2.png">

	<!-- Facebook OpenGraph -->
	<meta property="og:title" content="Scripts – DIY.transcriptomics" />
	<meta property="og:description" content="Access to all the scripts and code used throughout the course." />
	<meta property="og:image" content="http://localhost:4000/images/code2.png" />

	
	<!-- Font Embed Code -->
	<link href="https://fonts.googleapis.com/css?family=Muli:300,400,600,700" rel="stylesheet">
	

	<!-- Styles -->
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="/css/style.css">
	
	<!-- Icons -->
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/solid.js" integrity="sha384-GXi56ipjsBwAe6v5X4xSrVNXGOmpdJYZEEh/0/GqJ3JTHsfDsF8v0YQvZCJYAiGu" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/brands.js" integrity="sha384-0inRy4HkP0hJ038ZyfQ4vLl+F4POKbqnaUB6ewmU4dWP0ki8Q27A0VFiVRIpscvL" crossorigin="anonymous"></script>
	<script defer src="https://use.fontawesome.com/releases/v5.1.1/js/fontawesome.js" integrity="sha384-NY6PHjYLP2f+gL3uaVfqUZImmw71ArL9+Roi9o+I4+RBqArA2CfW1sJ1wkABFfPe" crossorigin="anonymous"></script>

	
	<!-- Custom Styles -->
	<style></style>
	

	
	<!-- Analytics Code -->
	
	

	
	<!-- Extra Header JS Code -->
	
	
	
</head>


<body class="loading ajax-loading" data-site-url="http://localhost:4000" data-page-url="/scripts">


	<header class="header">

	<div class="wrap">

		
		<a href="/" class="header__title">
			DIY.transcriptomics
		</a>
		

		<div class="menu">
			<div class="menu__toggle js-menu-toggle">
				<div class="menu__toggle__icon"><span></span></div>
			</div>
			<div class="menu__wrap">
				<ul class="menu__list">
					
					<li class="menu__list__item">
						<a href="/" class="menu__list__item__link">Syllabus</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/about" class="menu__list__item__link">About</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/data" class="menu__list__item__link">Data</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/scripts" class="menu__list__item__link">Scripts</a>
					</li>
					
					<li class="menu__list__item">
						<a href="/help" class="menu__list__item__link">Help</a>
					</li>
					
				</ul>
			</div>
		</div>

	</div>

</header>


	<div class="loader"><svg width="120" height="30" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg"><circle cx="15" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="60" cy="15" r="9" fill-opacity="0.3"><animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="0.5" to="0.5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite" /></circle><circle cx="105" cy="15" r="15"><animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite" /><animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite" /></circle></svg></div>

	<div class="page-loader"></div>

	
	<div class="page">

		<div class="page__content" data-page-title="Scripts – DIY.transcriptomics">

			<section class="hero hero--single">

	<div class="hero__image" style="background-image: url(/images/code2.png)">
		<div class="hero__overlay"></div>
	</div>

	<div class="wrap">

		<h1>Scripts</h1>
		<p>Access to all the scripts and code used throughout the course.</p>

	</div>

</section>

<section class="single">

	<div class="wrap">

		<article class="single-post">

			<h2 id="scripts">Scripts</h2>

<table>
  <thead>
    <tr>
      <th>Script Name</th>
      <th style="text-align: center">Packages</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Step1_Sleuth_TxImport.R">Step1_Sleuth_TxImport.R</a></td>
      <td style="text-align: center"><a href="http://bioconductor.org/packages/release/bioc/html/tximport.html">tximport</a>, <a href="https://www.tidyverse.org/">tidyverse</a>, <a href="https://github.com/pachterlab/sleuth">Sleuth</a>, <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">Biomart</a></td>
    </tr>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Step2_dataWrangling.R">Step2_dataWrangling.R</a></td>
      <td style="text-align: center"><a href="http://ggplot2.org/">ggplot2</a>, <a href="http://had.co.nz/reshape/">reshape2</a></td>
    </tr>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Step3_multivariate.R">Step3_multivariate.R</a></td>
      <td style="text-align: center"><a href="http://ggplot2.org/">ggplot2</a>, <a href="http://had.co.nz/reshape/">reshape2</a>, <a href="http://genomicsclass.github.io/book/pages/dplyr_tutorial.html">dplyr</a>, <a href="http://ggvis.rstudio.com/">ggvis</a>, <a href="https://github.com/juba/scatterD3">scatterD3</a></td>
    </tr>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Step4_publicData.R">Step4_publicData.R</a></td>
      <td style="text-align: center"><a href="http://bioconductor.org/packages/release/bioc/html/rhdf5.html">rhdf5</a>, <a href="https://github.com/VanAndelInstitute/slinky">slinky</a></td>
    </tr>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Step5_diffGenes.R">Step5_diffGenes.R</a></td>
      <td style="text-align: center"><a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a>, <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a>, <a href="https://bioconductor.org/packages/release/bioc/html/sva.html">SVA</a></td>
    </tr>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Step6_modules.R">Step6_modules.R</a></td>
      <td style="text-align: center"><a href="https://cran.r-project.org/web/packages/gplots/index.html">gplots</a>, <a href="http://earlglynn.github.io/RNotes/package/RColorBrewer/index.html">Rcolorbrewer</a>, <a href="https://cran.r-project.org/web/packages/heatmaply/index.html">heatmaply</a>, <a href="https://cran.r-project.org/web/packages/d3heatmap/index.html">d3heatmap</a></td>
    </tr>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Step7_functionalEnrichment.R">Step7_functionalEnrichment.R</a></td>
      <td style="text-align: center"><a href="http://bioconductor.org/packages/release/bioc/vignettes/GSVA/inst/doc/GSVA.pdf">GSVA</a>, <a href="http://bioconductor.org/packages/release/bioc/html/GSEABase.html">GSEAbase</a></td>
    </tr>
    <tr>
      <td><a href="http://DIYtranscriptomics.github.io/Code/files/Rmarkdown_example.Rmd">Rmarkdown_example.Rmd</a>, <a href="http://DIYtranscriptomics.github.io/Code/files/Txi_gene">Txi_gene</a>, <a href="http://DIYtranscriptomics.github.io/Code/files/myDGEList">myDGEList</a></td>
      <td style="text-align: center"><a href="http://rmarkdown.rstudio.com/">Rmarkdown</a>, <a href="http://yihui.name/knitr/">knitr</a></td>
    </tr>
  </tbody>
</table>

<hr />

<h2 id="data">Data</h2>

<p class="message">
You'll get the most from this class if you B.Y.O.D. -- Bring your own data.  Nothing will keep you more enagaged and crystalize the content of the course quite like working on questions you *actually* care about.  That said, it is not a requirement that you come with data in hand.  Throughout the course, I will demonstrate every step in the analysis using a 'real' dataset, which you can download below and use to follow along with me in class.  If you run into serious problems following along with your own data, I strongly recommend that you use the dataset provided below.  You can always return to your data once you have a functional pipeline and understand the process.  
</p>

<h3 id="the-response-of-the-intestinal-epithelium-to-infection-with-the-protozoan-parasite-cryptosporidium-parvum">The response of the intestinal epithelium to infection with the protozoan parasite, <em>Cryptosporidium parvum</em></h3>
<ul>
  <li>The featured dataset for this iteration of the course comes courtesy of <a href="http://www.striepenlab.org/">Boris Striepen’s lab</a> and is unpublished, so please be respectful of this.</li>
  <li>Download the <a href="https://www.dropbox.com/sh/df58trgab010s55/AAAQ86KkKPzuqvGG-YoeISNEa?dl=0">raw data</a>, which consists of 9 fastq files.  You will need about 30Gb of storage space on your harddrive to accomodate these file.  <em>please do not uncompress these files (leave them as .gz files)</em></li>
  <li>you also need to download this <a href="http://DIYtranscriptomics.github.io/Data/files/MIDAS.zip">basic project directory</a> that you can use to organize this (or any other) RNAseq project.  This project directory comes preloaded with some QC data, as well as a study design file, and a shell script for automating read mapping.  We’ll discuss these in more detail during class.</li>
  <li>In the event that you have any problems installing or using Kallisto to map this raw data, I’ve already mapped this data to produce transcript-level abundance data, which you can download and start working with immediately.  These files are available as a single compressed file <a href="https://drive.google.com/file/d/1KDbXsGT0EGW9qiVihesWvspYiYxSwW76/view?usp=sharing">here</a>.  Unzipping this file will reveal 9 folders (each containing the Kallisto output from mapping each of the 9 fastq files above), along with 9 log files (one for each sample that was mapped).  You may notice that each folder contains several files.  Please leave these in place.  During the course, we will discuss what these files actually mean.</li>
</ul>


		</article>

	</div>

</section>

		</div>

	</div>


	<footer class="footer">

	<div class="wrap">

		<p class="footer__text">A course on using open-source tools to analyze RNAseq data.</p>

		<div class="footer__copyright">
			<span>© 2019 DIY.transcriptomics</span>
			<a href="https://jekyllthemes.io" target="_blank">Jekyll Themes</a>
		</div>

		<ul class="socials">
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
</ul>

	</div>

</footer>


	<!-- Javascript Assets -->
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="/js/personal-min.js"></script>

	
	<!-- Extra Footer JS Code -->
	
	


</body>

</html>