---
title: 'Creating and using shell scripts and loops'
subtitle: 'Lab 2 (required) • January 31, 2024'
date: 2019-01-10 00:00:00
description: Using command-line tools often requires that you run similar code for each of your samples (e.g. read mapping).  In this lab, you'll learn how to automate this redundant process using a simple code-aware text editor, making it possible for you to get work done even when you're not sitting in front of your computer.  How great is that?!
featured_image: '/images/shellscript.png'
---

## Corresponding lecture

[Lecture 2 - Ultra-fast read mapping with Kallisto](https://diytranscriptomics.com/project/lecture-02)

## If you're new to R

Please take time to work through [this Learn R! module on basic R](https://diytranscriptomics.com/learnr/module-02)


## Description

In our lecture, we covered the basics for running Kallisto with a single sample.  However, you will rarely, if ever, be dealing with a single sample.  In this lab, we'll work through how to automate alignments and other command-line work using shell scripts and 'for loops'.  To keep the lab moving along, we'll work with very small fastq files (see below)

## Files you'll need for this lab

**[subsampled fastq files](https://drive.google.com/drive/folders/1RVzpLQH62yirGjGpIyjuunOLDlZi2Zjb?usp=sharing)** - This is the [course dataset](https://diytranscriptomics.com/data), but each file has been subsampled to retrieve only 10,000 reads per sample.

**[shell script](https://drive.google.com/file/d/1sGkMk-ekZk3pVD0aT5qzvMBo3UYnZSyU/view?usp=sharing)** – To carry out read mapping and QC analysis for multiple samples.

## Follow along

**[Bash basics](https://protocols.hostmicrobe.org/bash)** - this is the code I'll be running and discussing in lab today.  You can copy/paste lines from this page to follow along.

## By the end of this lab...

You should be able to run a for loop, using your bash program, to create a shell script.  You should then be able to run this shell script in bash to iterate over the subsampled fastq files (linked above) to quality check these reads, map to the human reference transcriptome, and produce a multiQC output.