
Class   |   Date    |   Topic   |
:------:|:---------:|-----------|
1   |   9/30    |   __Introduction to RNAseq data and technology__ ([slides](http://bit.ly/1OEOK7i))
2   |   10/14   |   __Setting up your software environment: an introduction to [R](http://www.r-project.org/), [RStudio](http://www.rstudio.com/), and [Git](http://git-scm.com/)__ ([slides](http://bit.ly/1W2veWM))
3   |   10/21   |   __Version control with Git, and [GitHub](https://github.com/)__ ([slides](http://bit.ly/1i1ljOl))
4   |   10/28   |   __Understanding normalization and variance in RNAseq data__ ([slides](http://bit.ly/1Reigie))
    |           |   * _Read alignment and summarization_
    |           |   * _Normalization and sources of variation_
    |           |   * _Looking to the literature for cases where they got it wrong_
5   |   11/04   |   __Starting your analysis script__
    |           |   * _defining your experiment in a simple study design file_
    |           |   * _raw data and normalization_
    |           |   * _annotation_
6   |   11/11   |   __Exploratory analysis of expression data (gene-agnostic)__
    |           |   * _viewing sample relationships with dist and hclust functions_ 
    |           |   * _using the prcomp function to carry out a principal component analysis_
    |           |   * _using [ggplot2](http://ggplot2.org/) to visualize your PCA analysis_
    |           |   * _using PCA 'loadings' to examine the relationship between samples and principal components_
7   |   11/18   |   __Open Working Lab - exploring your own data__
8   |   11/24   |   __Managing and tidying data in R__
    |       |   * _using [dplyr's](http://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html) mutate, sort and filter commands to edit data tables_
    |       |   * _static and interactive scatter plots using ggplot2 and [ggvis](http://ggvis.rstudio.com/)_
    |       |   * _melting data with the [reshape2](http://www.seananderson.ca/2013/10/19/reshape.html) package_
9 | 12/02 | __Identifying differentially expressed genes__
  |     | * _using the [Limma](http://www.bioconductor.org/packages/release/bioc/html/limma.html) package_
  |     | * _P values, false discovery rates, and volcano plots_
10 | 12/09  | __Visualizing and dissecting your differentially expressed gene list__
11 | 12/16  | __Understanding and leveraging [Gene Ontology](http://geneontology.org/) for functional enrichment analyses__ ([slides](http://bit.ly/1RO6r5V))
12 | 01/06  | __Gene Set Enrichment Analysis ([GSEA](http://www.broadinstitute.org/gsea/index.jsp)) and gene [signatures](http://www.broadinstitute.org/gsea/msigdb/index.jsp)__
13 | 01/13  | __Gene Set Enrichment Analysis-part II__
14  | 01/20 | __Open lab exercise - work on analysis summary figure__
15  | 01/27  | __Making your anaysis pipeline transparent and reproducible using [R Markdown](http://rmarkdown.rstudio.com/) and [Knitr](http://yihui.name/knitr/)__ ([slides](http://bit.ly/20teLJX))
16  | 02/03 | __Rmarkdown wrap-up__
17  | 02/10 | __Deploying your data to the web using [Shiny](https://shiny.rstudio.com)__ 
18  | 02/17 | __Final figure, Rmarkdown report, and shiny web app due__ 